{
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "XUZS4GUkHIEG"
      },
      "source": [
        "# ÐŸÐ¾Ð´Ð³Ð¾Ñ‚Ð¾Ð²ÐºÐ° Ñ„Ð°Ð¹Ð»Ð¾Ð² Ð¸ Ð¿Ñ€Ð¾Ð³Ñ€Ð°Ð¼Ð¼\n",
        "\n",
        "ÐÐµÐºÐ¾Ñ‚Ð¾Ñ€Ñ‹Ðµ Ð¿Ñ€Ð¾Ð³Ñ€Ð°Ð¼Ð¼Ñ‹ ÑÑ‚Ð°Ð²ÑÑ‚ÑÑ Ð´Ð²Ð°Ð¶Ð´Ñ‹ Ð¸Ð· Ñ€Ð°Ð·Ð½Ñ‹Ñ… Ð¸ÑÑ‚Ð¾Ñ‡Ð½Ð¸ÐºÐ¾Ð² Ð´Ð»Ñ Ð´ÐµÐ¼Ð¾Ð½ÑÑ‚Ñ€Ð°Ñ†Ð¸Ð¸ Ñ€Ð°Ð·Ð»Ð¸Ñ‡Ð½Ñ‹Ñ… Ð²Ð°Ñ€Ð¸Ð°Ð½Ñ‚Ð¾Ð² Ð² ÑÐ»ÑƒÑ‡Ð°Ðµ Ð½ÐµÐ¿Ð¾Ð»Ð°Ð´Ð¾Ðº. Ð”Ð»Ñ Ð¾Ð¿Ñ‚Ð¸Ð¼Ð¸Ð·Ð°Ñ†Ð¸Ð¸ Ð²Ñ€ÐµÐ¼ÐµÐ½Ð¸ Ñ€Ð°Ð±Ð¾Ñ‚Ñ‹ Ð½ÐµÐ¾Ð±Ñ…Ð¾Ð´Ð¸Ð¼Ð¾ ÑƒÐ´Ð°Ð»Ð¸Ñ‚ÑŒ Ð»Ð¸ÑˆÐ½Ð¸Ð¹ ÐºÐ¾Ð´."
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!pip install macs2"
      ],
      "metadata": {
        "id": "pV47sHC0DdT1",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "69ce21da-e700-402a-bdaf-b4a587b30067"
      },
      "execution_count": 1,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/\n",
            "Collecting macs2\n",
            "  Using cached MACS2-2.2.7.1.tar.gz (2.1 MB)\n",
            "  Installing build dependencies ... \u001b[?25l\u001b[?25hdone\n",
            "  Getting requirements to build wheel ... \u001b[?25l\u001b[?25hdone\n",
            "  \u001b[1;31merror\u001b[0m: \u001b[1msubprocess-exited-with-error\u001b[0m\n",
            "  \n",
            "  \u001b[31mÃ—\u001b[0m \u001b[32mPreparing metadata \u001b[0m\u001b[1;32m(\u001b[0m\u001b[32mpyproject.toml\u001b[0m\u001b[1;32m)\u001b[0m did not run successfully.\n",
            "  \u001b[31mâ”‚\u001b[0m exit code: \u001b[1;36m1\u001b[0m\n",
            "  \u001b[31mâ•°â”€>\u001b[0m See above for output.\n",
            "  \n",
            "  \u001b[1;35mnote\u001b[0m: This error originates from a subprocess, and is likely not a problem with pip.\n",
            "  Preparing metadata (pyproject.toml) ... \u001b[?25l\u001b[?25herror\n",
            "\u001b[1;31merror\u001b[0m: \u001b[1mmetadata-generation-failed\u001b[0m\n",
            "\n",
            "\u001b[31mÃ—\u001b[0m Encountered error while generating package metadata.\n",
            "\u001b[31mâ•°â”€>\u001b[0m See above for output.\n",
            "\n",
            "\u001b[1;35mnote\u001b[0m: This is an issue with the package mentioned above, not pip.\n",
            "\u001b[1;36mhint\u001b[0m: See above for details.\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 2,
      "metadata": {
        "id": "lrtRbP3fAvj8",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "cf1620d2-4748-4e6a-c761-70b827692107"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\u001b[33mWARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv\u001b[0m\u001b[33m\n",
            "\u001b[0m"
          ]
        }
      ],
      "source": [
        "pip install -q condacolab"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 3,
      "metadata": {
        "id": "jQ1Oqzb9BXoG"
      },
      "outputs": [],
      "source": [
        "import condacolab"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 4,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "q5FUOHCgBZWp",
        "outputId": "3d34f913-c39f-4cc3-d969-cc175ed82066"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "âœ¨ðŸ°âœ¨ Everything looks OK!\n"
          ]
        }
      ],
      "source": [
        "condacolab.install()"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 5,
      "metadata": {
        "id": "XnbIt77YBOT9",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "22aa971d-827a-4137-f4ba-e7d87c5c2875"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\bdone\n",
            "Solving environment: / \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bdone\n",
            "\n",
            "# All requested packages already installed.\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!conda install -c bioconda seqtk"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 6,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "sCBV_31qHOik",
        "outputId": "b39c2ab4-63a3-4eb5-9352-705d84237527"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\bdone\n",
            "Solving environment: \\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\bdone\n",
            "\n",
            "# All requested packages already installed.\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!conda install -c bioconda trimmomatic"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 7,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "F0BLWsH2n3Ov",
        "outputId": "e1f8058c-4816-4c20-8f84-77579fbbb041"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/\n",
            "Requirement already satisfied: intervene in /usr/local/lib/python3.8/site-packages (0.6.5)\n",
            "Requirement already satisfied: numpy in /usr/local/lib/python3.8/site-packages (from intervene) (1.24.2)\n",
            "Requirement already satisfied: seaborn in /usr/local/lib/python3.8/site-packages (from intervene) (0.12.2)\n",
            "Requirement already satisfied: pybedtools in /usr/local/lib/python3.8/site-packages (from intervene) (0.9.0)\n",
            "Requirement already satisfied: scipy in /usr/local/lib/python3.8/site-packages (from intervene) (1.10.1)\n",
            "Requirement already satisfied: pandas in /usr/local/lib/python3.8/site-packages (from intervene) (1.5.3)\n",
            "Requirement already satisfied: matplotlib in /usr/local/lib/python3.8/site-packages (from intervene) (3.7.0)\n",
            "Requirement already satisfied: contourpy>=1.0.1 in /usr/local/lib/python3.8/site-packages (from matplotlib->intervene) (1.0.7)\n",
            "Requirement already satisfied: fonttools>=4.22.0 in /usr/local/lib/python3.8/site-packages (from matplotlib->intervene) (4.38.0)\n",
            "Requirement already satisfied: pyparsing>=2.3.1 in /usr/local/lib/python3.8/site-packages (from matplotlib->intervene) (3.0.9)\n",
            "Requirement already satisfied: python-dateutil>=2.7 in /usr/local/lib/python3.8/site-packages (from matplotlib->intervene) (2.8.2)\n",
            "Requirement already satisfied: cycler>=0.10 in /usr/local/lib/python3.8/site-packages (from matplotlib->intervene) (0.11.0)\n",
            "Requirement already satisfied: importlib-resources>=3.2.0 in /usr/local/lib/python3.8/site-packages (from matplotlib->intervene) (5.12.0)\n",
            "Requirement already satisfied: pillow>=6.2.0 in /usr/local/lib/python3.8/site-packages (from matplotlib->intervene) (9.4.0)\n",
            "Requirement already satisfied: kiwisolver>=1.0.1 in /usr/local/lib/python3.8/site-packages (from matplotlib->intervene) (1.4.4)\n",
            "Requirement already satisfied: packaging>=20.0 in /usr/local/lib/python3.8/site-packages (from matplotlib->intervene) (23.0)\n",
            "Requirement already satisfied: pytz>=2020.1 in /usr/local/lib/python3.8/site-packages (from pandas->intervene) (2022.7.1)\n",
            "Requirement already satisfied: six in /usr/local/lib/python3.8/site-packages (from pybedtools->intervene) (1.16.0)\n",
            "Requirement already satisfied: pysam in /usr/local/lib/python3.8/site-packages (from pybedtools->intervene) (0.20.0)\n",
            "Requirement already satisfied: zipp>=3.1.0 in /usr/local/lib/python3.8/site-packages (from importlib-resources>=3.2.0->matplotlib->intervene) (3.15.0)\n",
            "\u001b[33mWARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv\u001b[0m\u001b[33m\n",
            "\u001b[0m"
          ]
        }
      ],
      "source": [
        "!pip install intervene"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 8,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Ce6hiSTdkarU",
        "outputId": "d237f0bb-e501-40e1-97f4-3e2aaf941d63"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Reading package lists... Done\n",
            "Building dependency tree       \n",
            "Reading state information... Done\n",
            "The following NEW packages will be installed:\n",
            "  bedtools\n",
            "0 upgraded, 1 newly installed, 0 to remove and 19 not upgraded.\n",
            "Need to get 580 kB of archives.\n",
            "After this operation, 2,055 kB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu focal/universe amd64 bedtools amd64 2.27.1+dfsg-4ubuntu1 [580 kB]\n",
            "Fetched 580 kB in 1s (902 kB/s)\n",
            "Selecting previously unselected package bedtools.\n",
            "(Reading database ... 128208 files and directories currently installed.)\n",
            "Preparing to unpack .../bedtools_2.27.1+dfsg-4ubuntu1_amd64.deb ...\n",
            "Unpacking bedtools (2.27.1+dfsg-4ubuntu1) ...\n",
            "Setting up bedtools (2.27.1+dfsg-4ubuntu1) ...\n"
          ]
        }
      ],
      "source": [
        "!apt-get install bedtools"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 9,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "AUML3dvsqT9E",
        "outputId": "8ef6f037-e039-4497-904d-b30d86fab7e4"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Reading package lists... Done\n",
            "Building dependency tree       \n",
            "Reading state information... Done\n",
            "The following additional packages will be installed:\n",
            "  cwltool fonts-glyphicons-halflings libc-ares2 libhts3 libjs-bootstrap\n",
            "  libnode64 nodejs nodejs-doc python3-bs4 python3-cachecontrol\n",
            "  python3-coloredlogs python3-dateutil python3-decorator python3-html5lib\n",
            "  python3-humanfriendly python3-isodate python3-lockfile python3-lxml\n",
            "  python3-mistune python3-msgpack python3-mypy-extensions python3-networkx\n",
            "  python3-prov python3-psutil python3-pyparsing python3-rdflib\n",
            "  python3-rdflib-jsonld python3-ruamel.yaml python3-schema-salad\n",
            "  python3-shellescape python3-soupsieve python3-sparqlwrapper\n",
            "  python3-typing-extensions python3-webencodings python3-yaml\n",
            "Suggested packages:\n",
            "  bowtie2-examples docker.io | singularity-container npm python3-genshi\n",
            "  python-lockfile-doc python3-lxml-dbg python-lxml-doc python-networkx-doc\n",
            "  python3-matplotlib python3-pygraphviz | python3-pydot python3-scipy\n",
            "  python-prov-doc python3-pydot python-psutil-doc python-pyparsing-doc\n",
            "  python-rdflib-doc\n",
            "The following NEW packages will be installed:\n",
            "  bowtie2 cwltool fonts-glyphicons-halflings libc-ares2 libhts3\n",
            "  libjs-bootstrap libnode64 nodejs nodejs-doc python3-bs4 python3-cachecontrol\n",
            "  python3-coloredlogs python3-dateutil python3-decorator python3-html5lib\n",
            "  python3-humanfriendly python3-isodate python3-lockfile python3-lxml\n",
            "  python3-mistune python3-msgpack python3-mypy-extensions python3-networkx\n",
            "  python3-prov python3-psutil python3-pyparsing python3-rdflib\n",
            "  python3-rdflib-jsonld python3-ruamel.yaml python3-schema-salad\n",
            "  python3-shellescape python3-soupsieve python3-sparqlwrapper\n",
            "  python3-typing-extensions python3-webencodings python3-yaml samtools\n",
            "0 upgraded, 37 newly installed, 0 to remove and 19 not upgraded.\n",
            "Need to get 13.5 MB of archives.\n",
            "After this operation, 64.0 MB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu focal-updates/main amd64 python3-yaml amd64 5.3.1-1ubuntu0.1 [136 kB]\n",
            "Get:2 http://archive.ubuntu.com/ubuntu focal/universe amd64 python3-humanfriendly all 4.18-2 [60.8 kB]\n",
            "Get:3 http://archive.ubuntu.com/ubuntu focal/universe amd64 python3-coloredlogs all 7.3-2 [28.0 kB]\n",
            "Get:4 http://archive.ubuntu.com/ubuntu focal/universe amd64 python3-mypy-extensions all 0.4.3-1 [5,436 B]\n",
            "Get:5 http://archive.ubuntu.com/ubuntu focal/main amd64 python3-dateutil all 2.7.3-3ubuntu1 [63.3 kB]\n",
            "Get:6 http://archive.ubuntu.com/ubuntu focal-updates/main amd64 python3-lxml amd64 4.5.0-1ubuntu0.5 [1,384 kB]\n",
            "Get:7 http://archive.ubuntu.com/ubuntu focal/main amd64 python3-decorator all 4.4.2-0ubuntu1 [10.3 kB]\n",
            "Get:8 http://archive.ubuntu.com/ubuntu focal/main amd64 python3-networkx all 2.4-3ubuntu1 [855 kB]\n",
            "Get:9 http://archive.ubuntu.com/ubuntu focal/universe amd64 python3-isodate all 0.6.0-2 [24.0 kB]\n",
            "Get:10 http://archive.ubuntu.com/ubuntu focal/main amd64 python3-pyparsing all 2.4.6-1 [61.3 kB]\n",
            "Get:11 http://archive.ubuntu.com/ubuntu focal/universe amd64 python3-rdflib all 4.2.2-5 [241 kB]\n",
            "Get:12 http://archive.ubuntu.com/ubuntu focal/universe amd64 python3-prov all 1.5.2-2 [92.8 kB]\n",
            "Get:13 http://archive.ubuntu.com/ubuntu focal/main amd64 python3-psutil amd64 5.5.1-1ubuntu4 [141 kB]\n",
            "Get:14 http://archive.ubuntu.com/ubuntu focal/universe amd64 python3-ruamel.yaml amd64 0.15.89-3build1 [228 kB]\n",
            "Get:15 http://archive.ubuntu.com/ubuntu focal/main amd64 python3-msgpack amd64 0.6.2-1 [73.5 kB]\n",
            "Get:16 http://archive.ubuntu.com/ubuntu focal/main amd64 python3-lockfile all 1:0.12.2-2ubuntu2 [14.6 kB]\n",
            "Get:17 http://archive.ubuntu.com/ubuntu focal/universe amd64 python3-cachecontrol all 0.12.6-1ubuntu2 [16.7 kB]\n",
            "Get:18 http://archive.ubuntu.com/ubuntu focal/universe amd64 python3-mistune all 0.8.4-2 [14.3 kB]\n",
            "Get:19 http://archive.ubuntu.com/ubuntu focal/universe amd64 python3-rdflib-jsonld all 0.4.0-5 [12.4 kB]\n",
            "Get:20 http://archive.ubuntu.com/ubuntu focal/universe amd64 python3-typing-extensions all 3.7.4.1-1 [27.5 kB]\n",
            "Get:21 http://archive.ubuntu.com/ubuntu focal/universe amd64 fonts-glyphicons-halflings all 1.009~3.4.1+dfsg-1 [117 kB]\n",
            "Get:22 http://archive.ubuntu.com/ubuntu focal/universe amd64 libjs-bootstrap all 3.4.1+dfsg-1 [124 kB]\n",
            "Get:23 http://archive.ubuntu.com/ubuntu focal/universe amd64 python3-schema-salad all 5.0.20200220195218-1 [295 kB]\n",
            "Get:24 http://archive.ubuntu.com/ubuntu focal/universe amd64 python3-shellescape all 3.4.1-4 [4,928 B]\n",
            "Get:25 http://archive.ubuntu.com/ubuntu focal/universe amd64 cwltool all 2.0.20200224214940+dfsg-1 [203 kB]\n",
            "Get:26 http://archive.ubuntu.com/ubuntu focal-updates/universe amd64 libhts3 amd64 1.10.2-3ubuntu0.1 [350 kB]\n",
            "Get:27 http://archive.ubuntu.com/ubuntu focal-updates/main amd64 libc-ares2 amd64 1.15.0-1ubuntu0.1 [38.2 kB]\n",
            "Get:28 http://archive.ubuntu.com/ubuntu focal/universe amd64 libnode64 amd64 10.19.0~dfsg-3ubuntu1 [5,765 kB]\n",
            "Get:29 http://archive.ubuntu.com/ubuntu focal/universe amd64 nodejs-doc all 10.19.0~dfsg-3ubuntu1 [942 kB]\n",
            "Get:30 http://archive.ubuntu.com/ubuntu focal/main amd64 python3-soupsieve all 1.9.5+dfsg-1 [29.1 kB]\n",
            "Get:31 http://archive.ubuntu.com/ubuntu focal/main amd64 python3-bs4 all 4.8.2-1 [83.0 kB]\n",
            "Get:32 http://archive.ubuntu.com/ubuntu focal/main amd64 python3-webencodings all 0.5.1-1ubuntu1 [11.0 kB]\n",
            "Get:33 http://archive.ubuntu.com/ubuntu focal/main amd64 python3-html5lib all 1.0.1-2 [84.3 kB]\n",
            "Get:34 http://archive.ubuntu.com/ubuntu focal/universe amd64 python3-sparqlwrapper all 1.8.5-1 [32.3 kB]\n",
            "Get:35 http://archive.ubuntu.com/ubuntu focal/universe amd64 samtools amd64 1.10-3 [496 kB]\n",
            "Get:36 http://archive.ubuntu.com/ubuntu focal/universe amd64 bowtie2 amd64 2.3.5.1-6build1 [1,324 kB]\n",
            "Get:37 http://archive.ubuntu.com/ubuntu focal/universe amd64 nodejs amd64 10.19.0~dfsg-3ubuntu1 [61.1 kB]\n",
            "Fetched 13.5 MB in 3s (4,252 kB/s)\n",
            "Extracting templates from packages: 100%\n",
            "Selecting previously unselected package python3-yaml.\n",
            "(Reading database ... 128261 files and directories currently installed.)\n",
            "Preparing to unpack .../00-python3-yaml_5.3.1-1ubuntu0.1_amd64.deb ...\n",
            "Unpacking python3-yaml (5.3.1-1ubuntu0.1) ...\n",
            "Selecting previously unselected package python3-humanfriendly.\n",
            "Preparing to unpack .../01-python3-humanfriendly_4.18-2_all.deb ...\n",
            "Unpacking python3-humanfriendly (4.18-2) ...\n",
            "Selecting previously unselected package python3-coloredlogs.\n",
            "Preparing to unpack .../02-python3-coloredlogs_7.3-2_all.deb ...\n",
            "Unpacking python3-coloredlogs (7.3-2) ...\n",
            "Selecting previously unselected package python3-mypy-extensions.\n",
            "Preparing to unpack .../03-python3-mypy-extensions_0.4.3-1_all.deb ...\n",
            "Unpacking python3-mypy-extensions (0.4.3-1) ...\n",
            "Selecting previously unselected package python3-dateutil.\n",
            "Preparing to unpack .../04-python3-dateutil_2.7.3-3ubuntu1_all.deb ...\n",
            "Unpacking python3-dateutil (2.7.3-3ubuntu1) ...\n",
            "Selecting previously unselected package python3-lxml:amd64.\n",
            "Preparing to unpack .../05-python3-lxml_4.5.0-1ubuntu0.5_amd64.deb ...\n",
            "Unpacking python3-lxml:amd64 (4.5.0-1ubuntu0.5) ...\n",
            "Selecting previously unselected package python3-decorator.\n",
            "Preparing to unpack .../06-python3-decorator_4.4.2-0ubuntu1_all.deb ...\n",
            "Unpacking python3-decorator (4.4.2-0ubuntu1) ...\n",
            "Selecting previously unselected package python3-networkx.\n",
            "Preparing to unpack .../07-python3-networkx_2.4-3ubuntu1_all.deb ...\n",
            "Unpacking python3-networkx (2.4-3ubuntu1) ...\n",
            "Selecting previously unselected package python3-isodate.\n",
            "Preparing to unpack .../08-python3-isodate_0.6.0-2_all.deb ...\n",
            "Unpacking python3-isodate (0.6.0-2) ...\n",
            "Selecting previously unselected package python3-pyparsing.\n",
            "Preparing to unpack .../09-python3-pyparsing_2.4.6-1_all.deb ...\n",
            "Unpacking python3-pyparsing (2.4.6-1) ...\n",
            "Selecting previously unselected package python3-rdflib.\n",
            "Preparing to unpack .../10-python3-rdflib_4.2.2-5_all.deb ...\n",
            "Unpacking python3-rdflib (4.2.2-5) ...\n",
            "Selecting previously unselected package python3-prov.\n",
            "Preparing to unpack .../11-python3-prov_1.5.2-2_all.deb ...\n",
            "Unpacking python3-prov (1.5.2-2) ...\n",
            "Selecting previously unselected package python3-psutil.\n",
            "Preparing to unpack .../12-python3-psutil_5.5.1-1ubuntu4_amd64.deb ...\n",
            "Unpacking python3-psutil (5.5.1-1ubuntu4) ...\n",
            "Selecting previously unselected package python3-ruamel.yaml.\n",
            "Preparing to unpack .../13-python3-ruamel.yaml_0.15.89-3build1_amd64.deb ...\n",
            "Unpacking python3-ruamel.yaml (0.15.89-3build1) ...\n",
            "Selecting previously unselected package python3-msgpack.\n",
            "Preparing to unpack .../14-python3-msgpack_0.6.2-1_amd64.deb ...\n",
            "Unpacking python3-msgpack (0.6.2-1) ...\n",
            "Selecting previously unselected package python3-lockfile.\n",
            "Preparing to unpack .../15-python3-lockfile_1%3a0.12.2-2ubuntu2_all.deb ...\n",
            "Unpacking python3-lockfile (1:0.12.2-2ubuntu2) ...\n",
            "Selecting previously unselected package python3-cachecontrol.\n",
            "Preparing to unpack .../16-python3-cachecontrol_0.12.6-1ubuntu2_all.deb ...\n",
            "Unpacking python3-cachecontrol (0.12.6-1ubuntu2) ...\n",
            "Selecting previously unselected package python3-mistune.\n",
            "Preparing to unpack .../17-python3-mistune_0.8.4-2_all.deb ...\n",
            "Unpacking python3-mistune (0.8.4-2) ...\n",
            "Selecting previously unselected package python3-rdflib-jsonld.\n",
            "Preparing to unpack .../18-python3-rdflib-jsonld_0.4.0-5_all.deb ...\n",
            "Unpacking python3-rdflib-jsonld (0.4.0-5) ...\n",
            "Selecting previously unselected package python3-typing-extensions.\n",
            "Preparing to unpack .../19-python3-typing-extensions_3.7.4.1-1_all.deb ...\n",
            "Unpacking python3-typing-extensions (3.7.4.1-1) ...\n",
            "Selecting previously unselected package fonts-glyphicons-halflings.\n",
            "Preparing to unpack .../20-fonts-glyphicons-halflings_1.009~3.4.1+dfsg-1_all.deb ...\n",
            "Unpacking fonts-glyphicons-halflings (1.009~3.4.1+dfsg-1) ...\n",
            "Selecting previously unselected package libjs-bootstrap.\n",
            "Preparing to unpack .../21-libjs-bootstrap_3.4.1+dfsg-1_all.deb ...\n",
            "Unpacking libjs-bootstrap (3.4.1+dfsg-1) ...\n",
            "Selecting previously unselected package python3-schema-salad.\n",
            "Preparing to unpack .../22-python3-schema-salad_5.0.20200220195218-1_all.deb ...\n",
            "Unpacking python3-schema-salad (5.0.20200220195218-1) ...\n",
            "Selecting previously unselected package python3-shellescape.\n",
            "Preparing to unpack .../23-python3-shellescape_3.4.1-4_all.deb ...\n",
            "Unpacking python3-shellescape (3.4.1-4) ...\n",
            "Selecting previously unselected package cwltool.\n",
            "Preparing to unpack .../24-cwltool_2.0.20200224214940+dfsg-1_all.deb ...\n",
            "Unpacking cwltool (2.0.20200224214940+dfsg-1) ...\n",
            "Selecting previously unselected package libhts3:amd64.\n",
            "Preparing to unpack .../25-libhts3_1.10.2-3ubuntu0.1_amd64.deb ...\n",
            "Unpacking libhts3:amd64 (1.10.2-3ubuntu0.1) ...\n",
            "Selecting previously unselected package libc-ares2:amd64.\n",
            "Preparing to unpack .../26-libc-ares2_1.15.0-1ubuntu0.1_amd64.deb ...\n",
            "Unpacking libc-ares2:amd64 (1.15.0-1ubuntu0.1) ...\n",
            "Selecting previously unselected package libnode64:amd64.\n",
            "Preparing to unpack .../27-libnode64_10.19.0~dfsg-3ubuntu1_amd64.deb ...\n",
            "Unpacking libnode64:amd64 (10.19.0~dfsg-3ubuntu1) ...\n",
            "Selecting previously unselected package nodejs-doc.\n",
            "Preparing to unpack .../28-nodejs-doc_10.19.0~dfsg-3ubuntu1_all.deb ...\n",
            "Unpacking nodejs-doc (10.19.0~dfsg-3ubuntu1) ...\n",
            "Selecting previously unselected package python3-soupsieve.\n",
            "Preparing to unpack .../29-python3-soupsieve_1.9.5+dfsg-1_all.deb ...\n",
            "Unpacking python3-soupsieve (1.9.5+dfsg-1) ...\n",
            "Selecting previously unselected package python3-bs4.\n",
            "Preparing to unpack .../30-python3-bs4_4.8.2-1_all.deb ...\n",
            "Unpacking python3-bs4 (4.8.2-1) ...\n",
            "Selecting previously unselected package python3-webencodings.\n",
            "Preparing to unpack .../31-python3-webencodings_0.5.1-1ubuntu1_all.deb ...\n",
            "Unpacking python3-webencodings (0.5.1-1ubuntu1) ...\n",
            "Selecting previously unselected package python3-html5lib.\n",
            "Preparing to unpack .../32-python3-html5lib_1.0.1-2_all.deb ...\n",
            "Unpacking python3-html5lib (1.0.1-2) ...\n",
            "Selecting previously unselected package python3-sparqlwrapper.\n",
            "Preparing to unpack .../33-python3-sparqlwrapper_1.8.5-1_all.deb ...\n",
            "Unpacking python3-sparqlwrapper (1.8.5-1) ...\n",
            "Selecting previously unselected package samtools.\n",
            "Preparing to unpack .../34-samtools_1.10-3_amd64.deb ...\n",
            "Unpacking samtools (1.10-3) ...\n",
            "Selecting previously unselected package bowtie2.\n",
            "Preparing to unpack .../35-bowtie2_2.3.5.1-6build1_amd64.deb ...\n",
            "Unpacking bowtie2 (2.3.5.1-6build1) ...\n",
            "Selecting previously unselected package nodejs.\n",
            "Preparing to unpack .../36-nodejs_10.19.0~dfsg-3ubuntu1_amd64.deb ...\n",
            "Unpacking nodejs (10.19.0~dfsg-3ubuntu1) ...\n",
            "Setting up python3-lockfile (1:0.12.2-2ubuntu2) ...\n",
            "Setting up libhts3:amd64 (1.10.2-3ubuntu0.1) ...\n",
            "Setting up fonts-glyphicons-halflings (1.009~3.4.1+dfsg-1) ...\n",
            "Setting up python3-mypy-extensions (0.4.3-1) ...\n",
            "Setting up python3-yaml (5.3.1-1ubuntu0.1) ...\n",
            "Setting up libc-ares2:amd64 (1.15.0-1ubuntu0.1) ...\n",
            "Setting up python3-webencodings (0.5.1-1ubuntu1) ...\n",
            "Setting up python3-psutil (5.5.1-1ubuntu4) ...\n",
            "Setting up samtools (1.10-3) ...\n",
            "Setting up bowtie2 (2.3.5.1-6build1) ...\n",
            "Setting up python3-isodate (0.6.0-2) ...\n",
            "Setting up python3-decorator (4.4.2-0ubuntu1) ...\n",
            "Setting up python3-pyparsing (2.4.6-1) ...\n",
            "Setting up python3-humanfriendly (4.18-2) ...\n",
            "Setting up libnode64:amd64 (10.19.0~dfsg-3ubuntu1) ...\n",
            "Setting up python3-typing-extensions (3.7.4.1-1) ...\n",
            "Setting up python3-html5lib (1.0.1-2) ...\n",
            "Setting up nodejs-doc (10.19.0~dfsg-3ubuntu1) ...\n",
            "Setting up python3-ruamel.yaml (0.15.89-3build1) ...\n",
            "Setting up python3-lxml:amd64 (4.5.0-1ubuntu0.5) ...\n",
            "Setting up python3-rdflib (4.2.2-5) ...\n",
            "/usr/lib/python3/dist-packages/rdflib/plugins/memory.py:354: SyntaxWarning: \"is\" with a literal. Did you mean \"==\"?\n",
            "  if triple is None or triple is (None,None,None):\n",
            "Setting up python3-dateutil (2.7.3-3ubuntu1) ...\n",
            "Setting up python3-msgpack (0.6.2-1) ...\n",
            "Setting up python3-cachecontrol (0.12.6-1ubuntu2) ...\n",
            "Setting up python3-shellescape (3.4.1-4) ...\n",
            "Setting up python3-coloredlogs (7.3-2) ...\n",
            "Setting up python3-mistune (0.8.4-2) ...\n",
            "Setting up python3-soupsieve (1.9.5+dfsg-1) ...\n",
            "Setting up libjs-bootstrap (3.4.1+dfsg-1) ...\n",
            "Setting up python3-rdflib-jsonld (0.4.0-5) ...\n",
            "Setting up python3-sparqlwrapper (1.8.5-1) ...\n",
            "Setting up python3-schema-salad (5.0.20200220195218-1) ...\n",
            "Setting up nodejs (10.19.0~dfsg-3ubuntu1) ...\n",
            "update-alternatives: using /usr/bin/nodejs to provide /usr/bin/js (js) in auto mode\n",
            "Setting up python3-networkx (2.4-3ubuntu1) ...\n",
            "/usr/lib/python3/dist-packages/networkx/readwrite/tests/test_graphml.py:933: SyntaxWarning: \"is\" with a literal. Did you mean \"==\"?\n",
            "  assert H.nodes['n1']['special'] is 0\n",
            "/usr/lib/python3/dist-packages/networkx/readwrite/tests/test_graphml.py:935: SyntaxWarning: \"is\" with a literal. Did you mean \"==\"?\n",
            "  assert H.edges['n0','n1',1]['special'] is 0\n",
            "Setting up python3-bs4 (4.8.2-1) ...\n",
            "Setting up python3-prov (1.5.2-2) ...\n",
            "Setting up cwltool (2.0.20200224214940+dfsg-1) ...\n",
            "/usr/lib/python3/dist-packages/cwltool/docker.py:423: SyntaxWarning: \"is\" with a literal. Did you mean \"==\"?\n",
            "  if res_req is not None and (\"ramMin\" in res_req or \"ramMax\" is res_req):\n",
            "Processing triggers for fontconfig (2.13.1-2ubuntu3) ...\n",
            "Processing triggers for libc-bin (2.31-0ubuntu9.9) ...\n",
            "Processing triggers for man-db (2.9.1-1) ...\n"
          ]
        }
      ],
      "source": [
        "!apt-get install bowtie2 samtools"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 10,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "JgTq4lFjq93N",
        "outputId": "f01afa8d-53f6-4bfc-eee3-d7fb2a356009"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Collecting package metadata (current_repodata.json): - \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bdone\n",
            "Solving environment: | \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\bdone\n",
            "\n",
            "## Package Plan ##\n",
            "\n",
            "  environment location: /usr/local\n",
            "\n",
            "  added / updated specs:\n",
            "    - bowtie2\n",
            "\n",
            "\n",
            "The following packages will be downloaded:\n",
            "\n",
            "    package                    |            build\n",
            "    ---------------------------|-----------------\n",
            "    bowtie2-2.5.1              |   py38h77f66f0_0        14.8 MB  bioconda\n",
            "    libhwloc-2.9.0             |       hd6dc26d_0         2.4 MB  conda-forge\n",
            "    perl-5.32.1                | 2_h7f98852_perl5        14.4 MB  conda-forge\n",
            "    tbb-2021.8.0               |       hf52228f_0         1.4 MB  conda-forge\n",
            "    ------------------------------------------------------------\n",
            "                                           Total:        33.1 MB\n",
            "\n",
            "The following NEW packages will be INSTALLED:\n",
            "\n",
            "  bowtie2            bioconda/linux-64::bowtie2-2.5.1-py38h77f66f0_0 \n",
            "  libhwloc           conda-forge/linux-64::libhwloc-2.9.0-hd6dc26d_0 \n",
            "  perl               conda-forge/linux-64::perl-5.32.1-2_h7f98852_perl5 \n",
            "  tbb                conda-forge/linux-64::tbb-2021.8.0-hf52228f_0 \n",
            "\n",
            "\n",
            "\n",
            "Downloading and Extracting Packages\n",
            "libhwloc-2.9.0       | 2.4 MB    | :   0% 0/1 [00:00<?, ?it/s]\n",
            "bowtie2-2.5.1        | 14.8 MB   | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\n",
            "\n",
            "tbb-2021.8.0         | 1.4 MB    | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "perl-5.32.1          | 14.4 MB   | :   0% 0/1 [00:00<?, ?it/s]\u001b[A\u001b[A\u001b[A\n",
            "libhwloc-2.9.0       | 2.4 MB    | :   1% 0.0063839528373015484/1 [00:00<01:24, 84.63s/it]\n",
            "\n",
            "tbb-2021.8.0         | 1.4 MB    | :   1% 0.010792658304727915/1 [00:00<00:49, 49.62s/it]\u001b[A\u001b[A\n",
            "\n",
            "\n",
            "perl-5.32.1          | 14.4 MB   | :   0% 0.0010852662651499435/1 [00:00<08:31, 512.22s/it]\u001b[A\u001b[A\u001b[A\n",
            "libhwloc-2.9.0       | 2.4 MB    | :  41% 0.4085729815872991/1 [00:00<00:00,  1.18s/it]   \n",
            "\n",
            "tbb-2021.8.0         | 1.4 MB    | :  84% 0.8418273477687773/1 [00:00<00:00,  1.76it/s]  \u001b[A\u001b[A\n",
            "\n",
            "\n",
            "perl-5.32.1          | 14.4 MB   | :   9% 0.08573603494684554/1 [00:00<00:05,  5.76s/it]   \u001b[A\u001b[A\u001b[A\n",
            "bowtie2-2.5.1        | 14.8 MB   | :  19% 0.19163219488906544/1 [00:00<00:02,  2.49s/it]\u001b[A\n",
            "\n",
            "\n",
            "perl-5.32.1          | 14.4 MB   | :  22% 0.21813851929513864/1 [00:00<00:01,  2.31s/it]\u001b[A\u001b[A\u001b[A\n",
            "bowtie2-2.5.1        | 14.8 MB   | :  33% 0.3313860607750137/1 [00:00<00:01,  1.56s/it] \u001b[A\n",
            "\n",
            "\n",
            "perl-5.32.1          | 14.4 MB   | :  34% 0.3396883409919323/1 [00:00<00:01,  1.61s/it] \u001b[A\u001b[A\u001b[A\n",
            "bowtie2-2.5.1        | 14.8 MB   | :  46% 0.4637287368033738/1 [00:00<00:00,  1.22s/it]\u001b[A\n",
            "\n",
            "tbb-2021.8.0         | 1.4 MB    | : 100% 1.0/1 [00:00<00:00,  1.76it/s]               \u001b[A\u001b[A\n",
            "\n",
            "\n",
            "perl-5.32.1          | 14.4 MB   | :  47% 0.46992029280992553/1 [00:00<00:00,  1.25s/it]\u001b[A\u001b[A\u001b[A\n",
            "bowtie2-2.5.1        | 14.8 MB   | :  62% 0.6151287581798177/1 [00:01<00:00,  1.01s/it]\u001b[A\n",
            "\n",
            "\n",
            "perl-5.32.1          | 14.4 MB   | :  60% 0.6034080434233686/1 [00:01<00:00,  1.08s/it] \u001b[A\u001b[A\u001b[A\n",
            "bowtie2-2.5.1        | 14.8 MB   | :  77% 0.7665287795562618/1 [00:01<00:00,  1.11it/s]\u001b[A\n",
            "\n",
            "\n",
            "perl-5.32.1          | 14.4 MB   | :  77% 0.7727095807867598/1 [00:01<00:00,  1.12it/s]\u001b[A\u001b[A\u001b[A\n",
            "bowtie2-2.5.1        | 14.8 MB   | :  94% 0.9433385947301508/1 [00:01<00:00,  1.27it/s]\u001b[A\n",
            "\n",
            "\n",
            "libhwloc-2.9.0       | 2.4 MB    | : 100% 1.0/1 [00:01<00:00,  1.63s/it]\n",
            "bowtie2-2.5.1        | 14.8 MB   | : 100% 1.0/1 [00:08<00:00,  1.27it/s]               \u001b[A\n",
            "\n",
            "\n",
            "                                                                        \n",
            "                                                                        \u001b[A\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\n",
            "\n",
            "\n",
            "                                                                        \u001b[A\u001b[A\u001b[A\n",
            "Preparing transaction: | \b\b/ \b\bdone\n",
            "Verifying transaction: \\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\b- \b\b\\ \b\b| \b\b/ \b\bdone\n",
            "Executing transaction: \\ \b\b| \b\b/ \b\b- \b\bdone\n"
          ]
        }
      ],
      "source": [
        "!conda install -c bioconda bowtie2"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "vcr5CjibHTgF"
      },
      "source": [
        "## Ð£ÑÑ‚Ð°Ð½Ð¾Ð²ÐºÐ° FastQC"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 11,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "62l1Q2ymbENA",
        "outputId": "a63b0abf-9998-4322-f3ed-9baaed2281fe"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "openjdk version \"17.0.3-internal\" 2022-04-19\n",
            "OpenJDK Runtime Environment (build 17.0.3-internal+0-adhoc..src)\n",
            "OpenJDK 64-Bit Server VM (build 17.0.3-internal+0-adhoc..src, mixed mode, sharing)\n"
          ]
        }
      ],
      "source": [
        "!java -version"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 12,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "8bpgwFgcbgRd",
        "outputId": "c44a3488-44fd-4f5b-e160-42aece3094c0"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2023-03-01 18:11:27--  https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.9.zip\n",
            "Resolving www.bioinformatics.babraham.ac.uk (www.bioinformatics.babraham.ac.uk)... 149.155.133.4\n",
            "Connecting to www.bioinformatics.babraham.ac.uk (www.bioinformatics.babraham.ac.uk)|149.155.133.4|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 10249221 (9.8M) [application/zip]\n",
            "Saving to: â€˜fastqc_v0.11.9.zipâ€™\n",
            "\n",
            "fastqc_v0.11.9.zip  100%[===================>]   9.77M  2.00MB/s    in 4.9s    \n",
            "\n",
            "2023-03-01 18:11:33 (2.00 MB/s) - â€˜fastqc_v0.11.9.zipâ€™ saved [10249221/10249221]\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!wget https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.9.zip"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 13,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "yTcTjOTJblcm",
        "outputId": "6957ec07-b52a-4fc6-bcd9-92edf6e43399"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Archive:  fastqc_v0.11.9.zip\n",
            "  inflating: FastQC/cisd-jhdf5.jar   \n",
            "   creating: FastQC/Configuration/\n",
            "  inflating: FastQC/Configuration/adapter_list.txt  \n",
            "  inflating: FastQC/Configuration/contaminant_list.txt  \n",
            "  inflating: FastQC/Configuration/limits.txt  \n",
            "  inflating: FastQC/fastqc           \n",
            "  inflating: FastQC/fastqc_icon.ico  \n",
            "   creating: FastQC/Help/\n",
            "   creating: FastQC/Help/1 Introduction/\n",
            "   creating: FastQC/Help/1 Introduction/.svn/\n",
            "  inflating: FastQC/Help/1 Introduction/.svn/entries  \n",
            "   creating: FastQC/Help/1 Introduction/.svn/props/\n",
            "   creating: FastQC/Help/1 Introduction/.svn/text-base/\n",
            "  inflating: FastQC/Help/1 Introduction/.svn/text-base/1.1 What is FastQC.html.svn-base  \n",
            "   creating: FastQC/Help/1 Introduction/.svn/tmp/\n",
            "   creating: FastQC/Help/1 Introduction/.svn/tmp/props/\n",
            "  inflating: FastQC/Help/1 Introduction/1.1 What is FastQC.html  \n",
            "   creating: FastQC/Help/2 Basic Operations/\n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/\n",
            "  inflating: FastQC/Help/2 Basic Operations/.svn/entries  \n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/props/\n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/text-base/\n",
            "  inflating: FastQC/Help/2 Basic Operations/.svn/text-base/2.1 Opening a sequence file.html.svn-base  \n",
            "  inflating: FastQC/Help/2 Basic Operations/.svn/text-base/2.2 Evaluating Results.html.svn-base  \n",
            "  inflating: FastQC/Help/2 Basic Operations/.svn/text-base/2.3 Saving a Report.html.svn-base  \n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/tmp/\n",
            "   creating: FastQC/Help/2 Basic Operations/.svn/tmp/props/\n",
            "  inflating: FastQC/Help/2 Basic Operations/2.1 Opening a sequence file.html  \n",
            "  inflating: FastQC/Help/2 Basic Operations/2.2 Evaluating Results.html  \n",
            "  inflating: FastQC/Help/2 Basic Operations/2.3 Saving a Report.html  \n",
            "   creating: FastQC/Help/3 Analysis Modules/\n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/\n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/entries  \n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/prop-base/\n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/duplication_levels.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/prop-base/kmer_profiles.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_gc_content.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_n_content.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_quality.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_base_sequence_content.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_sequence_gc_content.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_sequence_quality.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/per_tile_quality.png.svn-base  \n",
            " extracting: FastQC/Help/3 Analysis Modules/.svn/prop-base/sequence_length_distribution.png.svn-base  \n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/props/\n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/text-base/\n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/1 Basic Statistics.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/10 Adapter Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/11 Kmer Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/12 Per Tile Sequence Quality.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/2 Per Base Sequence Quality.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/3 Per Sequence Quality Scores.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/4 Per Base Sequence Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/5 Per Sequence GC Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/6 Per Base N Content.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/7 Sequence Length Distribution.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/8 Duplicate Sequences.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/9 Overrepresented Sequences.html.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/duplication_levels.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/kmer_profiles.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_gc_content.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_n_content.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_quality.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_base_sequence_content.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_sequence_gc_content.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_sequence_quality.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/per_tile_quality.png.svn-base  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/.svn/text-base/sequence_length_distribution.png.svn-base  \n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/tmp/\n",
            "   creating: FastQC/Help/3 Analysis Modules/.svn/tmp/props/\n",
            "  inflating: FastQC/Help/3 Analysis Modules/1 Basic Statistics.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/10 Adapter Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/11 Kmer Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/12 Per Tile Sequence Quality.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/2 Per Base Sequence Quality.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/3 Per Sequence Quality Scores.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/4 Per Base Sequence Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/5 Per Sequence GC Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/6 Per Base N Content.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/7 Sequence Length Distribution.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/8 Duplicate Sequences.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/9 Overrepresented Sequences.html  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/duplication_levels.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/kmer_profiles.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_base_gc_content.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_base_n_content.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_base_quality.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_base_sequence_content.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_sequence_gc_content.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_sequence_quality.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/per_tile_quality.png  \n",
            "  inflating: FastQC/Help/3 Analysis Modules/sequence_length_distribution.png  \n",
            "  inflating: FastQC/INSTALL.txt      \n",
            "  inflating: FastQC/jbzip2-0.9.jar   \n",
            "  inflating: FastQC/LICENSE          \n",
            "  inflating: FastQC/LICENSE.txt      \n",
            "  inflating: FastQC/LICENSE_JHDF5.txt  \n",
            "   creating: FastQC/net/\n",
            "   creating: FastQC/net/sourceforge/\n",
            "   creating: FastQC/net/sourceforge/iharder/\n",
            "   creating: FastQC/net/sourceforge/iharder/base64/\n",
            "  inflating: FastQC/net/sourceforge/iharder/base64/Base64$1.class  \n",
            "  inflating: FastQC/net/sourceforge/iharder/base64/Base64$InputStream.class  \n",
            "  inflating: FastQC/net/sourceforge/iharder/base64/Base64$OutputStream.class  \n",
            "  inflating: FastQC/net/sourceforge/iharder/base64/Base64.class  \n",
            "   creating: FastQC/org/\n",
            "   creating: FastQC/org/apache/\n",
            "   creating: FastQC/org/apache/commons/\n",
            "   creating: FastQC/org/apache/commons/math3/\n",
            "   creating: FastQC/org/apache/commons/math3/analysis/\n",
            "   creating: FastQC/org/apache/commons/math3/analysis/solvers/\n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/AbstractUnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/AllowedSolution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BaseAbstractUnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BaseUnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BracketedUnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/BrentSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/UnivariateSolver.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/solvers/UnivariateSolverUtils.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/analysis/UnivariateFunction.class  \n",
            "   creating: FastQC/org/apache/commons/math3/distribution/\n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/AbstractIntegerDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/AbstractRealDistribution$1.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/AbstractRealDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/BetaDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/BinomialDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/CauchyDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/ChiSquaredDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/FDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/GammaDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/HypergeometricDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/IntegerDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/NormalDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/PascalDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/PoissonDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/RealDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/SaddlePointExpansion.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/TDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/WeibullDistribution.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/distribution/ZipfDistribution.class  \n",
            "   creating: FastQC/org/apache/commons/math3/exception/\n",
            "  inflating: FastQC/org/apache/commons/math3/exception/ConvergenceException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/DimensionMismatchException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathArithmeticException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathIllegalArgumentException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathIllegalNumberException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathIllegalStateException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MathInternalError.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/MaxCountExceededException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NoBracketingException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NotFiniteNumberException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NotPositiveException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NotStrictlyPositiveException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NullArgumentException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NumberIsTooLargeException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/NumberIsTooSmallException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/OutOfRangeException.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/TooManyEvaluationsException.class  \n",
            "   creating: FastQC/org/apache/commons/math3/exception/util/\n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/ArgUtils.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/ExceptionContext.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/ExceptionContextProvider.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/Localizable.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/exception/util/LocalizedFormats.class  \n",
            "   creating: FastQC/org/apache/commons/math3/random/\n",
            "  inflating: FastQC/org/apache/commons/math3/random/AbstractWell.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/BitsStreamGenerator.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/RandomData.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/RandomDataImpl.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/RandomGenerator.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/random/Well19937c.class  \n",
            "   creating: FastQC/org/apache/commons/math3/special/\n",
            "  inflating: FastQC/org/apache/commons/math3/special/Beta$1.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/special/Beta.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/special/Erf.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/special/Gamma$1.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/special/Gamma.class  \n",
            "   creating: FastQC/org/apache/commons/math3/util/\n",
            "  inflating: FastQC/org/apache/commons/math3/util/ArithmeticUtils.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/ContinuedFraction.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/DoubleArray.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMath$ExpFracTable.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMath$ExpIntTable.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMath$lnMant.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMath.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMathCalc.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/FastMathLiteralArrays.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/Incrementor$1.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/Incrementor$MaxCountExceededCallback.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/Incrementor.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/MathUtils.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/Precision.class  \n",
            "  inflating: FastQC/org/apache/commons/math3/util/ResizableDoubleArray.class  \n",
            "  inflating: FastQC/README.md        \n",
            "  inflating: FastQC/README.txt       \n",
            "  inflating: FastQC/RELEASE_NOTES.txt  \n",
            "  inflating: FastQC/run_fastqc.bat   \n",
            "  inflating: FastQC/sam-1.103.jar    \n",
            "   creating: FastQC/Templates/\n",
            "  inflating: FastQC/Templates/fastqc2fo.xsl  \n",
            "  inflating: FastQC/Templates/header_template.html  \n",
            "   creating: FastQC/Templates/Icons/\n",
            " extracting: FastQC/Templates/Icons/error.png  \n",
            " extracting: FastQC/Templates/Icons/fastqc_icon.png  \n",
            " extracting: FastQC/Templates/Icons/tick.png  \n",
            " extracting: FastQC/Templates/Icons/warning.png  \n",
            "   creating: FastQC/uk/\n",
            "   creating: FastQC/uk/ac/\n",
            "   creating: FastQC/uk/ac/babraham/\n",
            "   creating: FastQC/uk/ac/babraham/FastQC/\n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Analysis/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/AnalysisListener.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/AnalysisQueue.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/AnalysisRunner.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Analysis/OfflineRunner.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Dialogs/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/AboutDialog$1.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/AboutDialog.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/FastQCTitlePanel$SmoothJLabel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/FastQCTitlePanel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Dialogs/WelcomePanel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCApplication$1.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCApplication.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCConfig.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FastQCMenuBar.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/FileFilters/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/BAMFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/CasavaFastQFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/FastQFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/GobyFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/MappedBAMFileFilter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/FileFilters/SequenceFileFilter.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Graphs/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/BaseGroup.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/LineGraph.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/QualityBoxPlot.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Graphs/TileGraph.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Help/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpDialog.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpIndexRoot$FileSorter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpIndexRoot.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpPage.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpPageDisplay$HelpEditor.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpPageDisplay.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Help/HelpSearchPanel.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Modules/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AbstractQCModule.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AdapterContent$Adapter.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AdapterContent$ResultsTable.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/AdapterContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/BasicStats$ResultsTable.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/BasicStats.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/DuplicationLevel.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Modules/GCModel/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/GCModel/GCModel.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/GCModel/GCModelValue.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/KmerContent$Kmer.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/KmerContent$ResultsTable.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/KmerContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/ModuleConfig.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/ModuleFactory.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/NContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/OverRepresentedSeqs$OverrepresentedSeq.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/OverRepresentedSeqs$ResultsTable.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/OverRepresentedSeqs.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerBaseQualityScores.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerBaseSequenceContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerSequenceGCContent.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerSequenceQualityScores.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/PerTileQualityScores.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/QCModule.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Modules/SequenceLengthDistribution.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Report/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Report/HTMLReportArchive.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Report/stylesheet.txt  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Resources/\n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/error.png  \n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/fastqc_icon.png  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Resources/fastqc_icon.svg  \n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/fastqc_icon_100.png  \n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/tick.png  \n",
            " extracting: FastQC/uk/ac/babraham/FastQC/Resources/warning.png  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Results/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Results/ResultsPanel$ModuleRenderer.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Results/ResultsPanel.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Sequence/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/BAMFile.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/Contaminant.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/ContaminantHit.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Contaminant/ContaminentFinder.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Fast5File.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/FastQFile.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Sequence/QualityEncoding/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/QualityEncoding/PhredEncoding.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/Sequence.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFactory.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFile.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFileGroup.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Sequence/SequenceFormatException.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Statistics/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Statistics/NormalDistribution.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Statistics/PearsonCorrelation.class  \n",
            "   creating: FastQC/uk/ac/babraham/FastQC/Utilities/\n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/CasavaBasename.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/HotColdColourGradient.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/ImageToBase64.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/MultiMemberGZIPInputStream.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/NameFormatException.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/NanoporeBasename.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/QualityCount.class  \n",
            "  inflating: FastQC/uk/ac/babraham/FastQC/Utilities/RGB.class  \n"
          ]
        }
      ],
      "source": [
        "!unzip fastqc_v0.11.9.zip"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 14,
      "metadata": {
        "id": "S79LH0aXbvLx"
      },
      "outputs": [],
      "source": [
        "!chmod a+x FastQC/fastqc"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "PsLuZDt4b7jT",
        "outputId": "a1daea71-850c-4cff-f193-66344dedabd2"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "            FastQC - A high throughput sequence QC analysis tool\n",
            "\n",
            "SYNOPSIS\n",
            "\n",
            "\tfastqc seqfile1 seqfile2 .. seqfileN\n",
            "\n",
            "    fastqc [-o output dir] [--(no)extract] [-f fastq|bam|sam] \n",
            "           [-c contaminant file] seqfile1 .. seqfileN\n",
            "\n",
            "DESCRIPTION\n",
            "\n",
            "    FastQC reads a set of sequence files and produces from each one a quality\n",
            "    control report consisting of a number of different modules, each one of \n",
            "    which will help to identify a different potential type of problem in your\n",
            "    data.\n",
            "    \n",
            "    If no files to process are specified on the command line then the program\n",
            "    will start as an interactive graphical application.  If files are provided\n",
            "    on the command line then the program will run with no user interaction\n",
            "    required.  In this mode it is suitable for inclusion into a standardised\n",
            "    analysis pipeline.\n",
            "    \n",
            "    The options for the program as as follows:\n",
            "    \n",
            "    -h --help       Print this help file and exit\n",
            "    \n",
            "    -v --version    Print the version of the program and exit\n",
            "    \n",
            "    -o --outdir     Create all output files in the specified output directory.\n",
            "                    Please note that this directory must exist as the program\n",
            "                    will not create it.  If this option is not set then the \n",
            "                    output file for each sequence file is created in the same\n",
            "                    directory as the sequence file which was processed.\n",
            "                    \n",
            "    --casava        Files come from raw casava output. Files in the same sample\n",
            "                    group (differing only by the group number) will be analysed\n",
            "                    as a set rather than individually. Sequences with the filter\n",
            "                    flag set in the header will be excluded from the analysis.\n",
            "                    Files must have the same names given to them by casava\n",
            "                    (including being gzipped and ending with .gz) otherwise they\n",
            "                    won't be grouped together correctly.\n",
            "                    \n",
            "    --nano          Files come from nanopore sequences and are in fast5 format. In\n",
            "                    this mode you can pass in directories to process and the program\n",
            "                    will take in all fast5 files within those directories and produce\n",
            "                    a single output file from the sequences found in all files.                    \n",
            "                    \n",
            "    --nofilter      If running with --casava then don't remove read flagged by\n",
            "                    casava as poor quality when performing the QC analysis.\n",
            "                   \n",
            "    --extract       If set then the zipped output file will be uncompressed in\n",
            "                    the same directory after it has been created.  By default\n",
            "                    this option will be set if fastqc is run in non-interactive\n",
            "                    mode.\n",
            "                    \n",
            "    -j --java       Provides the full path to the java binary you want to use to\n",
            "                    launch fastqc. If not supplied then java is assumed to be in\n",
            "                    your path.\n",
            "                   \n",
            "    --noextract     Do not uncompress the output file after creating it.  You\n",
            "                    should set this option if you do not wish to uncompress\n",
            "                    the output when running in non-interactive mode.\n",
            "                    \n",
            "    --nogroup       Disable grouping of bases for reads >50bp. All reports will\n",
            "                    show data for every base in the read.  WARNING: Using this\n",
            "                    option will cause fastqc to crash and burn if you use it on\n",
            "                    really long reads, and your plots may end up a ridiculous size.\n",
            "                    You have been warned!\n",
            "                    \n",
            "    --min_length    Sets an artificial lower limit on the length of the sequence\n",
            "                    to be shown in the report.  As long as you set this to a value\n",
            "                    greater or equal to your longest read length then this will be\n",
            "                    the sequence length used to create your read groups.  This can\n",
            "                    be useful for making directly comaparable statistics from \n",
            "                    datasets with somewhat variable read lengths.\n",
            "                    \n",
            "    -f --format     Bypasses the normal sequence file format detection and\n",
            "                    forces the program to use the specified format.  Valid\n",
            "                    formats are bam,sam,bam_mapped,sam_mapped and fastq\n",
            "                    \n",
            "    -t --threads    Specifies the number of files which can be processed\n",
            "                    simultaneously.  Each thread will be allocated 250MB of\n",
            "                    memory so you shouldn't run more threads than your\n",
            "                    available memory will cope with, and not more than\n",
            "                    6 threads on a 32 bit machine\n",
            "                  \n",
            "    -c              Specifies a non-default file which contains the list of\n",
            "    --contaminants  contaminants to screen overrepresented sequences against.\n",
            "                    The file must contain sets of named contaminants in the\n",
            "                    form name[tab]sequence.  Lines prefixed with a hash will\n",
            "                    be ignored.\n",
            "\n",
            "    -a              Specifies a non-default file which contains the list of\n",
            "    --adapters      adapter sequences which will be explicity searched against\n",
            "                    the library. The file must contain sets of named adapters\n",
            "                    in the form name[tab]sequence.  Lines prefixed with a hash\n",
            "                    will be ignored.\n",
            "                    \n",
            "    -l              Specifies a non-default file which contains a set of criteria\n",
            "    --limits        which will be used to determine the warn/error limits for the\n",
            "                    various modules.  This file can also be used to selectively \n",
            "                    remove some modules from the output all together.  The format\n",
            "                    needs to mirror the default limits.txt file found in the\n",
            "                    Configuration folder.\n",
            "                    \n",
            "   -k --kmers       Specifies the length of Kmer to look for in the Kmer content\n",
            "                    module. Specified Kmer length must be between 2 and 10. Default\n",
            "                    length is 7 if not specified.\n",
            "                    \n",
            "   -q --quiet       Supress all progress messages on stdout and only report errors.\n",
            "   \n",
            "   -d --dir         Selects a directory to be used for temporary files written when\n",
            "                    generating report images. Defaults to system temp directory if\n",
            "                    not specified.\n",
            "                    \n",
            "BUGS\n",
            "\n",
            "    Any bugs in fastqc should be reported either to simon.andrews@babraham.ac.uk\n",
            "    or in www.bioinformatics.babraham.ac.uk/bugzilla/\n",
            "                   \n",
            "    \n"
          ]
        }
      ],
      "source": [
        "!./FastQC/fastqc --help"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "10Fs3qrmHmJi"
      },
      "source": [
        "# Ð’Ñ‹Ñ€Ð°Ð²Ð½Ð¸Ð²Ð°Ð½Ð¸Ðµ"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "-z4GQiMfU1wf"
      },
      "source": [
        "## Ð¡ÐºÐ°Ñ‡Ð¸Ð²Ð°ÐµÐ¼ Ñ‡Ñ‚ÐµÐ½Ð¸Ñ\n",
        "\n"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 15,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "lFWnMg0uU0sJ",
        "outputId": "36f1be0a-f553-4bd4-d92f-3da4381faefd"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2023-03-01 18:11:47--  https://www.encodeproject.org/files/ENCFF815VLD/@@download/ENCFF815VLD.fastq.gz\n",
            "Resolving www.encodeproject.org (www.encodeproject.org)... 34.211.244.144\n",
            "Connecting to www.encodeproject.org (www.encodeproject.org)|34.211.244.144|:443... connected.\n",
            "HTTP request sent, awaiting response... 307 Temporary Redirect\n",
            "Location: https://encode-public.s3.amazonaws.com/2014/10/24/5a3ad94c-7f9b-4704-b7f3-4f7c8ccaab0d/ENCFF815VLD.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF815VLD.fastq.gz&AWSAccessKeyId=ASIATGZNGCNX4NQMJBDH&Signature=Hq2oW4bEwV8YOKLB%2FtZAdDO%2BO3k%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEOr%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEaCXVzLXdlc3QtMiJGMEQCIA5FJb%2F5OJYPvKI0hF8NDOE%2Bd4j0KaOwFg9p3HQfoknPAiBx5RfEpEV%2FgTi19XtNgbix7Qf%2F39JMMVYEGo8zHw4Kmyq7BQiT%2F%2F%2F%2F%2F%2F%2F%2F%2F%2F8BEAAaDDIyMDc0ODcxNDg2MyIMvkIT3FA2zGV9aX5zKo8FsVMf4MJkaJJ0vU6KNZUq8AJ7%2BIydOuPUYSRp6ukO1K8zCwUA4gY59t12NTDej4%2Fiad9eB9tMdqF6ULS2KN42nEervM4HL%2Bj1GXV9p%2F%2BtNcUSGWPC3GWeBlm8j8X3tGmbhZ5DJJuVDxwyDi3zXCK82NNIW98lbdlxbzqe37n8IbyUOJxPJDCohxWQUlmg1OY%2Fl4PrGpZoGpM7DfElKknWON5G%2BXPCZJNpid9hlyxW3ASU5UWHmnR9Zaai6fSnoPLwrBLO2TiMXgSRStWLrD4YNDaZbMFRT7wisctMoUi9B1E7kuc50neTez%2BvBVUYR2pVq3dlAUPtBq%2Bb4vBTilRaJj0hTvUrMyxGZObWwrEJN0h1JahTaui%2BxdGMOl%2FWrifqBpPNulbleJt7KzqgT%2BhLuZfMCNxe%2F%2BgNaCALhT2vFR8EvjdbWPxjuGkb%2BAqBzF3qll5lyf9LQ%2B0DmdQRA%2FGjfmy7EILpFUxJkYmD2DmVWPnZHKIeDfgo%2FOhz3cy6j8Gbvrl556wYB3aparY5RMJandg1PwCiaJzE%2F68mNQ0zWND0rdAysGcuBFRDCz7HN2A5faDO99pglVdbASQC4a%2FG4j23E7Fvq6b4lb5Wk915K7eL6lg7i%2BD13t4cdzMi6g1Dfd6PlNCe4zSyC6ZNew7grH5%2BIY3rdUJaHQwEF2TP%2FqjqUd5miDU966%2BMJz1PBxOFUrvmcnH0INQO8QkjLXIrJ0%2B66CA1db4Rd8kRPoTbDFTCxFvoWGOJTvvQ20EP31ujfEjDm7scHjQZaHyaWf6l0ApaB3qE71b6EJ%2Bi8ob5GDNuYHsorizmF%2Bpn91%2FRlr2Hh1wKxmIZbJb9FJPCTWFk67opmYthYTdiN2wIU5vU%2FzCLof6fBjqyAUzYpQe2QSXItHnEl4mgWjQYP8Pgzzc6SGTZNUixCSTMsW82yVyyz1MtZFs%2FmB7EHcpWzmkfEgCYejiDGpN5l5bKIBg4BeEQRfo%2BM022mH%2BtMUw%2BdrCaN8GsZG8249ynOP8qR8YKBVFoexi1JZFVouGKwpCVerbZeIJT3qTvWiRZbpEJ8SfL0K0oRR4RGp0rQ1QCaXqEtO1PrBONvRN%2Fd%2BJndqjrMmlR%2F70KMc5nUDyiG%2Bw%3D&Expires=1677823907 [following]\n",
            "--2023-03-01 18:11:47--  https://encode-public.s3.amazonaws.com/2014/10/24/5a3ad94c-7f9b-4704-b7f3-4f7c8ccaab0d/ENCFF815VLD.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF815VLD.fastq.gz&AWSAccessKeyId=ASIATGZNGCNX4NQMJBDH&Signature=Hq2oW4bEwV8YOKLB%2FtZAdDO%2BO3k%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEOr%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEaCXVzLXdlc3QtMiJGMEQCIA5FJb%2F5OJYPvKI0hF8NDOE%2Bd4j0KaOwFg9p3HQfoknPAiBx5RfEpEV%2FgTi19XtNgbix7Qf%2F39JMMVYEGo8zHw4Kmyq7BQiT%2F%2F%2F%2F%2F%2F%2F%2F%2F%2F8BEAAaDDIyMDc0ODcxNDg2MyIMvkIT3FA2zGV9aX5zKo8FsVMf4MJkaJJ0vU6KNZUq8AJ7%2BIydOuPUYSRp6ukO1K8zCwUA4gY59t12NTDej4%2Fiad9eB9tMdqF6ULS2KN42nEervM4HL%2Bj1GXV9p%2F%2BtNcUSGWPC3GWeBlm8j8X3tGmbhZ5DJJuVDxwyDi3zXCK82NNIW98lbdlxbzqe37n8IbyUOJxPJDCohxWQUlmg1OY%2Fl4PrGpZoGpM7DfElKknWON5G%2BXPCZJNpid9hlyxW3ASU5UWHmnR9Zaai6fSnoPLwrBLO2TiMXgSRStWLrD4YNDaZbMFRT7wisctMoUi9B1E7kuc50neTez%2BvBVUYR2pVq3dlAUPtBq%2Bb4vBTilRaJj0hTvUrMyxGZObWwrEJN0h1JahTaui%2BxdGMOl%2FWrifqBpPNulbleJt7KzqgT%2BhLuZfMCNxe%2F%2BgNaCALhT2vFR8EvjdbWPxjuGkb%2BAqBzF3qll5lyf9LQ%2B0DmdQRA%2FGjfmy7EILpFUxJkYmD2DmVWPnZHKIeDfgo%2FOhz3cy6j8Gbvrl556wYB3aparY5RMJandg1PwCiaJzE%2F68mNQ0zWND0rdAysGcuBFRDCz7HN2A5faDO99pglVdbASQC4a%2FG4j23E7Fvq6b4lb5Wk915K7eL6lg7i%2BD13t4cdzMi6g1Dfd6PlNCe4zSyC6ZNew7grH5%2BIY3rdUJaHQwEF2TP%2FqjqUd5miDU966%2BMJz1PBxOFUrvmcnH0INQO8QkjLXIrJ0%2B66CA1db4Rd8kRPoTbDFTCxFvoWGOJTvvQ20EP31ujfEjDm7scHjQZaHyaWf6l0ApaB3qE71b6EJ%2Bi8ob5GDNuYHsorizmF%2Bpn91%2FRlr2Hh1wKxmIZbJb9FJPCTWFk67opmYthYTdiN2wIU5vU%2FzCLof6fBjqyAUzYpQe2QSXItHnEl4mgWjQYP8Pgzzc6SGTZNUixCSTMsW82yVyyz1MtZFs%2FmB7EHcpWzmkfEgCYejiDGpN5l5bKIBg4BeEQRfo%2BM022mH%2BtMUw%2BdrCaN8GsZG8249ynOP8qR8YKBVFoexi1JZFVouGKwpCVerbZeIJT3qTvWiRZbpEJ8SfL0K0oRR4RGp0rQ1QCaXqEtO1PrBONvRN%2Fd%2BJndqjrMmlR%2F70KMc5nUDyiG%2Bw%3D&Expires=1677823907\n",
            "Resolving encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)... 52.218.132.67, 52.218.133.177, 52.218.178.43, ...\n",
            "Connecting to encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)|52.218.132.67|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 1410910367 (1.3G) [binary/octet-stream]\n",
            "Saving to: â€˜ENCFF815VLD.fastq.gzâ€™\n",
            "\n",
            "ENCFF815VLD.fastq.g 100%[===================>]   1.31G  36.4MB/s    in 36s     \n",
            "\n",
            "2023-03-01 18:12:24 (37.1 MB/s) - â€˜ENCFF815VLD.fastq.gzâ€™ saved [1410910367/1410910367]\n",
            "\n",
            "--2023-03-01 18:12:24--  https://www.encodeproject.org/files/ENCFF644SSR/@@download/ENCFF644SSR.fastq.gz\n",
            "Resolving www.encodeproject.org (www.encodeproject.org)... 34.211.244.144\n",
            "Connecting to www.encodeproject.org (www.encodeproject.org)|34.211.244.144|:443... connected.\n",
            "HTTP request sent, awaiting response... 307 Temporary Redirect\n",
            "Location: https://encode-public.s3.amazonaws.com/2014/10/24/8fe8e1fb-7e05-474a-bd22-b00e6e0bb133/ENCFF644SSR.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF644SSR.fastq.gz&AWSAccessKeyId=ASIATGZNGCNX4NQMJBDH&Signature=HrJjkps1RsEOOmc8%2BJjnW1%2FQkFM%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEOr%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEaCXVzLXdlc3QtMiJGMEQCIA5FJb%2F5OJYPvKI0hF8NDOE%2Bd4j0KaOwFg9p3HQfoknPAiBx5RfEpEV%2FgTi19XtNgbix7Qf%2F39JMMVYEGo8zHw4Kmyq7BQiT%2F%2F%2F%2F%2F%2F%2F%2F%2F%2F8BEAAaDDIyMDc0ODcxNDg2MyIMvkIT3FA2zGV9aX5zKo8FsVMf4MJkaJJ0vU6KNZUq8AJ7%2BIydOuPUYSRp6ukO1K8zCwUA4gY59t12NTDej4%2Fiad9eB9tMdqF6ULS2KN42nEervM4HL%2Bj1GXV9p%2F%2BtNcUSGWPC3GWeBlm8j8X3tGmbhZ5DJJuVDxwyDi3zXCK82NNIW98lbdlxbzqe37n8IbyUOJxPJDCohxWQUlmg1OY%2Fl4PrGpZoGpM7DfElKknWON5G%2BXPCZJNpid9hlyxW3ASU5UWHmnR9Zaai6fSnoPLwrBLO2TiMXgSRStWLrD4YNDaZbMFRT7wisctMoUi9B1E7kuc50neTez%2BvBVUYR2pVq3dlAUPtBq%2Bb4vBTilRaJj0hTvUrMyxGZObWwrEJN0h1JahTaui%2BxdGMOl%2FWrifqBpPNulbleJt7KzqgT%2BhLuZfMCNxe%2F%2BgNaCALhT2vFR8EvjdbWPxjuGkb%2BAqBzF3qll5lyf9LQ%2B0DmdQRA%2FGjfmy7EILpFUxJkYmD2DmVWPnZHKIeDfgo%2FOhz3cy6j8Gbvrl556wYB3aparY5RMJandg1PwCiaJzE%2F68mNQ0zWND0rdAysGcuBFRDCz7HN2A5faDO99pglVdbASQC4a%2FG4j23E7Fvq6b4lb5Wk915K7eL6lg7i%2BD13t4cdzMi6g1Dfd6PlNCe4zSyC6ZNew7grH5%2BIY3rdUJaHQwEF2TP%2FqjqUd5miDU966%2BMJz1PBxOFUrvmcnH0INQO8QkjLXIrJ0%2B66CA1db4Rd8kRPoTbDFTCxFvoWGOJTvvQ20EP31ujfEjDm7scHjQZaHyaWf6l0ApaB3qE71b6EJ%2Bi8ob5GDNuYHsorizmF%2Bpn91%2FRlr2Hh1wKxmIZbJb9FJPCTWFk67opmYthYTdiN2wIU5vU%2FzCLof6fBjqyAUzYpQe2QSXItHnEl4mgWjQYP8Pgzzc6SGTZNUixCSTMsW82yVyyz1MtZFs%2FmB7EHcpWzmkfEgCYejiDGpN5l5bKIBg4BeEQRfo%2BM022mH%2BtMUw%2BdrCaN8GsZG8249ynOP8qR8YKBVFoexi1JZFVouGKwpCVerbZeIJT3qTvWiRZbpEJ8SfL0K0oRR4RGp0rQ1QCaXqEtO1PrBONvRN%2Fd%2BJndqjrMmlR%2F70KMc5nUDyiG%2Bw%3D&Expires=1677823944 [following]\n",
            "--2023-03-01 18:12:24--  https://encode-public.s3.amazonaws.com/2014/10/24/8fe8e1fb-7e05-474a-bd22-b00e6e0bb133/ENCFF644SSR.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF644SSR.fastq.gz&AWSAccessKeyId=ASIATGZNGCNX4NQMJBDH&Signature=HrJjkps1RsEOOmc8%2BJjnW1%2FQkFM%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEOr%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEaCXVzLXdlc3QtMiJGMEQCIA5FJb%2F5OJYPvKI0hF8NDOE%2Bd4j0KaOwFg9p3HQfoknPAiBx5RfEpEV%2FgTi19XtNgbix7Qf%2F39JMMVYEGo8zHw4Kmyq7BQiT%2F%2F%2F%2F%2F%2F%2F%2F%2F%2F8BEAAaDDIyMDc0ODcxNDg2MyIMvkIT3FA2zGV9aX5zKo8FsVMf4MJkaJJ0vU6KNZUq8AJ7%2BIydOuPUYSRp6ukO1K8zCwUA4gY59t12NTDej4%2Fiad9eB9tMdqF6ULS2KN42nEervM4HL%2Bj1GXV9p%2F%2BtNcUSGWPC3GWeBlm8j8X3tGmbhZ5DJJuVDxwyDi3zXCK82NNIW98lbdlxbzqe37n8IbyUOJxPJDCohxWQUlmg1OY%2Fl4PrGpZoGpM7DfElKknWON5G%2BXPCZJNpid9hlyxW3ASU5UWHmnR9Zaai6fSnoPLwrBLO2TiMXgSRStWLrD4YNDaZbMFRT7wisctMoUi9B1E7kuc50neTez%2BvBVUYR2pVq3dlAUPtBq%2Bb4vBTilRaJj0hTvUrMyxGZObWwrEJN0h1JahTaui%2BxdGMOl%2FWrifqBpPNulbleJt7KzqgT%2BhLuZfMCNxe%2F%2BgNaCALhT2vFR8EvjdbWPxjuGkb%2BAqBzF3qll5lyf9LQ%2B0DmdQRA%2FGjfmy7EILpFUxJkYmD2DmVWPnZHKIeDfgo%2FOhz3cy6j8Gbvrl556wYB3aparY5RMJandg1PwCiaJzE%2F68mNQ0zWND0rdAysGcuBFRDCz7HN2A5faDO99pglVdbASQC4a%2FG4j23E7Fvq6b4lb5Wk915K7eL6lg7i%2BD13t4cdzMi6g1Dfd6PlNCe4zSyC6ZNew7grH5%2BIY3rdUJaHQwEF2TP%2FqjqUd5miDU966%2BMJz1PBxOFUrvmcnH0INQO8QkjLXIrJ0%2B66CA1db4Rd8kRPoTbDFTCxFvoWGOJTvvQ20EP31ujfEjDm7scHjQZaHyaWf6l0ApaB3qE71b6EJ%2Bi8ob5GDNuYHsorizmF%2Bpn91%2FRlr2Hh1wKxmIZbJb9FJPCTWFk67opmYthYTdiN2wIU5vU%2FzCLof6fBjqyAUzYpQe2QSXItHnEl4mgWjQYP8Pgzzc6SGTZNUixCSTMsW82yVyyz1MtZFs%2FmB7EHcpWzmkfEgCYejiDGpN5l5bKIBg4BeEQRfo%2BM022mH%2BtMUw%2BdrCaN8GsZG8249ynOP8qR8YKBVFoexi1JZFVouGKwpCVerbZeIJT3qTvWiRZbpEJ8SfL0K0oRR4RGp0rQ1QCaXqEtO1PrBONvRN%2Fd%2BJndqjrMmlR%2F70KMc5nUDyiG%2Bw%3D&Expires=1677823944\n",
            "Resolving encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)... 52.92.192.209, 52.92.149.121, 52.218.185.171, ...\n",
            "Connecting to encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)|52.92.192.209|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 2415735698 (2.2G) [binary/octet-stream]\n",
            "Saving to: â€˜ENCFF644SSR.fastq.gzâ€™\n",
            "\n",
            "ENCFF644SSR.fastq.g 100%[===================>]   2.25G  43.3MB/s    in 58s     \n",
            "\n",
            "2023-03-01 18:13:23 (39.5 MB/s) - â€˜ENCFF644SSR.fastq.gzâ€™ saved [2415735698/2415735698]\n",
            "\n"
          ]
        }
      ],
      "source": [
        "# ChIP-seq Ð½Ð° Ð³Ð¸ÑÑ‚Ð¾Ð½Ð¾Ð²Ð¾Ð¹ Ð¼ÐµÑ‚ÐºÐµ\n",
        "!wget https://www.encodeproject.org/files/ENCFF815VLD/@@download/ENCFF815VLD.fastq.gz\n",
        "!wget https://www.encodeproject.org/files/ENCFF644SSR/@@download/ENCFF644SSR.fastq.gz\n"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# ChIP-seq ÐºÐ¾Ð½Ñ‚Ñ€Ð¾Ð»ÑŒ\n",
        "!wget https://www.encodeproject.org/files/ENCFF349YSG/@@download/ENCFF349YSG.fastq.gz"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "8XVCJRS7IeN6",
        "outputId": "d7bbeb30-3aba-41f4-f9e0-0606a780e5a0"
      },
      "execution_count": 16,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2023-03-01 18:13:23--  https://www.encodeproject.org/files/ENCFF349YSG/@@download/ENCFF349YSG.fastq.gz\n",
            "Resolving www.encodeproject.org (www.encodeproject.org)... 34.211.244.144\n",
            "Connecting to www.encodeproject.org (www.encodeproject.org)|34.211.244.144|:443... connected.\n",
            "HTTP request sent, awaiting response... 307 Temporary Redirect\n",
            "Location: https://encode-public.s3.amazonaws.com/2016/10/11/748be80d-5910-4b33-bc29-18e9d02c3bab/ENCFF349YSG.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF349YSG.fastq.gz&AWSAccessKeyId=ASIATGZNGCNX4NQMJBDH&Signature=fFJyVAmW1jNv0Y5jpo8MrHzpvKs%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEOr%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEaCXVzLXdlc3QtMiJGMEQCIA5FJb%2F5OJYPvKI0hF8NDOE%2Bd4j0KaOwFg9p3HQfoknPAiBx5RfEpEV%2FgTi19XtNgbix7Qf%2F39JMMVYEGo8zHw4Kmyq7BQiT%2F%2F%2F%2F%2F%2F%2F%2F%2F%2F8BEAAaDDIyMDc0ODcxNDg2MyIMvkIT3FA2zGV9aX5zKo8FsVMf4MJkaJJ0vU6KNZUq8AJ7%2BIydOuPUYSRp6ukO1K8zCwUA4gY59t12NTDej4%2Fiad9eB9tMdqF6ULS2KN42nEervM4HL%2Bj1GXV9p%2F%2BtNcUSGWPC3GWeBlm8j8X3tGmbhZ5DJJuVDxwyDi3zXCK82NNIW98lbdlxbzqe37n8IbyUOJxPJDCohxWQUlmg1OY%2Fl4PrGpZoGpM7DfElKknWON5G%2BXPCZJNpid9hlyxW3ASU5UWHmnR9Zaai6fSnoPLwrBLO2TiMXgSRStWLrD4YNDaZbMFRT7wisctMoUi9B1E7kuc50neTez%2BvBVUYR2pVq3dlAUPtBq%2Bb4vBTilRaJj0hTvUrMyxGZObWwrEJN0h1JahTaui%2BxdGMOl%2FWrifqBpPNulbleJt7KzqgT%2BhLuZfMCNxe%2F%2BgNaCALhT2vFR8EvjdbWPxjuGkb%2BAqBzF3qll5lyf9LQ%2B0DmdQRA%2FGjfmy7EILpFUxJkYmD2DmVWPnZHKIeDfgo%2FOhz3cy6j8Gbvrl556wYB3aparY5RMJandg1PwCiaJzE%2F68mNQ0zWND0rdAysGcuBFRDCz7HN2A5faDO99pglVdbASQC4a%2FG4j23E7Fvq6b4lb5Wk915K7eL6lg7i%2BD13t4cdzMi6g1Dfd6PlNCe4zSyC6ZNew7grH5%2BIY3rdUJaHQwEF2TP%2FqjqUd5miDU966%2BMJz1PBxOFUrvmcnH0INQO8QkjLXIrJ0%2B66CA1db4Rd8kRPoTbDFTCxFvoWGOJTvvQ20EP31ujfEjDm7scHjQZaHyaWf6l0ApaB3qE71b6EJ%2Bi8ob5GDNuYHsorizmF%2Bpn91%2FRlr2Hh1wKxmIZbJb9FJPCTWFk67opmYthYTdiN2wIU5vU%2FzCLof6fBjqyAUzYpQe2QSXItHnEl4mgWjQYP8Pgzzc6SGTZNUixCSTMsW82yVyyz1MtZFs%2FmB7EHcpWzmkfEgCYejiDGpN5l5bKIBg4BeEQRfo%2BM022mH%2BtMUw%2BdrCaN8GsZG8249ynOP8qR8YKBVFoexi1JZFVouGKwpCVerbZeIJT3qTvWiRZbpEJ8SfL0K0oRR4RGp0rQ1QCaXqEtO1PrBONvRN%2Fd%2BJndqjrMmlR%2F70KMc5nUDyiG%2Bw%3D&Expires=1677824003 [following]\n",
            "--2023-03-01 18:13:23--  https://encode-public.s3.amazonaws.com/2016/10/11/748be80d-5910-4b33-bc29-18e9d02c3bab/ENCFF349YSG.fastq.gz?response-content-disposition=attachment%3B%20filename%3DENCFF349YSG.fastq.gz&AWSAccessKeyId=ASIATGZNGCNX4NQMJBDH&Signature=fFJyVAmW1jNv0Y5jpo8MrHzpvKs%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEOr%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEaCXVzLXdlc3QtMiJGMEQCIA5FJb%2F5OJYPvKI0hF8NDOE%2Bd4j0KaOwFg9p3HQfoknPAiBx5RfEpEV%2FgTi19XtNgbix7Qf%2F39JMMVYEGo8zHw4Kmyq7BQiT%2F%2F%2F%2F%2F%2F%2F%2F%2F%2F8BEAAaDDIyMDc0ODcxNDg2MyIMvkIT3FA2zGV9aX5zKo8FsVMf4MJkaJJ0vU6KNZUq8AJ7%2BIydOuPUYSRp6ukO1K8zCwUA4gY59t12NTDej4%2Fiad9eB9tMdqF6ULS2KN42nEervM4HL%2Bj1GXV9p%2F%2BtNcUSGWPC3GWeBlm8j8X3tGmbhZ5DJJuVDxwyDi3zXCK82NNIW98lbdlxbzqe37n8IbyUOJxPJDCohxWQUlmg1OY%2Fl4PrGpZoGpM7DfElKknWON5G%2BXPCZJNpid9hlyxW3ASU5UWHmnR9Zaai6fSnoPLwrBLO2TiMXgSRStWLrD4YNDaZbMFRT7wisctMoUi9B1E7kuc50neTez%2BvBVUYR2pVq3dlAUPtBq%2Bb4vBTilRaJj0hTvUrMyxGZObWwrEJN0h1JahTaui%2BxdGMOl%2FWrifqBpPNulbleJt7KzqgT%2BhLuZfMCNxe%2F%2BgNaCALhT2vFR8EvjdbWPxjuGkb%2BAqBzF3qll5lyf9LQ%2B0DmdQRA%2FGjfmy7EILpFUxJkYmD2DmVWPnZHKIeDfgo%2FOhz3cy6j8Gbvrl556wYB3aparY5RMJandg1PwCiaJzE%2F68mNQ0zWND0rdAysGcuBFRDCz7HN2A5faDO99pglVdbASQC4a%2FG4j23E7Fvq6b4lb5Wk915K7eL6lg7i%2BD13t4cdzMi6g1Dfd6PlNCe4zSyC6ZNew7grH5%2BIY3rdUJaHQwEF2TP%2FqjqUd5miDU966%2BMJz1PBxOFUrvmcnH0INQO8QkjLXIrJ0%2B66CA1db4Rd8kRPoTbDFTCxFvoWGOJTvvQ20EP31ujfEjDm7scHjQZaHyaWf6l0ApaB3qE71b6EJ%2Bi8ob5GDNuYHsorizmF%2Bpn91%2FRlr2Hh1wKxmIZbJb9FJPCTWFk67opmYthYTdiN2wIU5vU%2FzCLof6fBjqyAUzYpQe2QSXItHnEl4mgWjQYP8Pgzzc6SGTZNUixCSTMsW82yVyyz1MtZFs%2FmB7EHcpWzmkfEgCYejiDGpN5l5bKIBg4BeEQRfo%2BM022mH%2BtMUw%2BdrCaN8GsZG8249ynOP8qR8YKBVFoexi1JZFVouGKwpCVerbZeIJT3qTvWiRZbpEJ8SfL0K0oRR4RGp0rQ1QCaXqEtO1PrBONvRN%2Fd%2BJndqjrMmlR%2F70KMc5nUDyiG%2Bw%3D&Expires=1677824003\n",
            "Resolving encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)... 52.92.194.105, 52.218.184.67, 52.92.130.17, ...\n",
            "Connecting to encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)|52.92.194.105|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 593415042 (566M) [binary/octet-stream]\n",
            "Saving to: â€˜ENCFF349YSG.fastq.gzâ€™\n",
            "\n",
            "ENCFF349YSG.fastq.g 100%[===================>] 565.92M  46.6MB/s    in 13s     \n",
            "\n",
            "2023-03-01 18:13:37 (42.3 MB/s) - â€˜ENCFF349YSG.fastq.gzâ€™ saved [593415042/593415042]\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 17,
      "metadata": {
        "id": "v73fH7WiVDu9"
      },
      "outputs": [],
      "source": [
        "# Ñ€Ð°ÑÐ¿Ð°ÐºÐ¾Ð²ÐºÐ° Ð°Ñ…Ñ€Ð¸Ð²Ð°\n",
        "!gzip -d /content/ENCFF815VLD.fastq.gz /content/ENCFF644SSR.fastq.gz"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#  ÐºÐ¾Ð½Ñ‚Ñ€Ð¾Ð»ÑŒ\n",
        "!gzip -d /content/ENCFF349YSG.fastq.gz"
      ],
      "metadata": {
        "id": "Kh7RDOFCIlaK"
      },
      "execution_count": 18,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "6sPcIefNHx-L"
      },
      "source": [
        "## FastQC (Ð¸Ð»Ð¸ multiQC Ð¼Ð¾Ð¶ÐµÑ‚ Ð±Ñ‹Ñ‚ÑŒ Ð¸ÑÐ¿Ð¾Ð»ÑŒÐ·Ð¾Ð²Ð°Ð½ Ð² Ñ€Ð°Ð±Ð¾Ñ‚Ðµ)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 19,
      "metadata": {
        "id": "5XM6jBcuGjMV",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "90732e6e-7fd7-4d4e-ddd2-45f021e08a84"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Started analysis of ENCFF815VLD.fastq\n",
            "Approx 5% complete for ENCFF815VLD.fastq\n",
            "Approx 10% complete for ENCFF815VLD.fastq\n",
            "Approx 15% complete for ENCFF815VLD.fastq\n",
            "Approx 20% complete for ENCFF815VLD.fastq\n",
            "Approx 25% complete for ENCFF815VLD.fastq\n",
            "Approx 30% complete for ENCFF815VLD.fastq\n",
            "Approx 35% complete for ENCFF815VLD.fastq\n",
            "Approx 40% complete for ENCFF815VLD.fastq\n",
            "Approx 45% complete for ENCFF815VLD.fastq\n",
            "Approx 50% complete for ENCFF815VLD.fastq\n",
            "Approx 55% complete for ENCFF815VLD.fastq\n",
            "Approx 60% complete for ENCFF815VLD.fastq\n",
            "Approx 65% complete for ENCFF815VLD.fastq\n",
            "Approx 70% complete for ENCFF815VLD.fastq\n",
            "Approx 75% complete for ENCFF815VLD.fastq\n",
            "Approx 80% complete for ENCFF815VLD.fastq\n",
            "Approx 85% complete for ENCFF815VLD.fastq\n",
            "Approx 90% complete for ENCFF815VLD.fastq\n",
            "Approx 95% complete for ENCFF815VLD.fastq\n",
            "Analysis complete for ENCFF815VLD.fastq\n",
            "Started analysis of ENCFF644SSR.fastq\n",
            "Approx 5% complete for ENCFF644SSR.fastq\n",
            "Approx 10% complete for ENCFF644SSR.fastq\n",
            "Approx 15% complete for ENCFF644SSR.fastq\n",
            "Approx 20% complete for ENCFF644SSR.fastq\n",
            "Approx 25% complete for ENCFF644SSR.fastq\n",
            "Approx 30% complete for ENCFF644SSR.fastq\n",
            "Approx 35% complete for ENCFF644SSR.fastq\n",
            "Approx 40% complete for ENCFF644SSR.fastq\n",
            "Approx 45% complete for ENCFF644SSR.fastq\n",
            "Approx 50% complete for ENCFF644SSR.fastq\n",
            "Approx 55% complete for ENCFF644SSR.fastq\n",
            "Approx 60% complete for ENCFF644SSR.fastq\n",
            "Approx 65% complete for ENCFF644SSR.fastq\n",
            "Approx 70% complete for ENCFF644SSR.fastq\n",
            "Approx 75% complete for ENCFF644SSR.fastq\n",
            "Approx 80% complete for ENCFF644SSR.fastq\n",
            "Approx 85% complete for ENCFF644SSR.fastq\n",
            "Approx 90% complete for ENCFF644SSR.fastq\n",
            "Approx 95% complete for ENCFF644SSR.fastq\n",
            "Analysis complete for ENCFF644SSR.fastq\n",
            "Started analysis of ENCFF349YSG.fastq\n",
            "Approx 5% complete for ENCFF349YSG.fastq\n",
            "Approx 10% complete for ENCFF349YSG.fastq\n",
            "Approx 15% complete for ENCFF349YSG.fastq\n",
            "Approx 20% complete for ENCFF349YSG.fastq\n",
            "Approx 25% complete for ENCFF349YSG.fastq\n",
            "Approx 30% complete for ENCFF349YSG.fastq\n",
            "Approx 35% complete for ENCFF349YSG.fastq\n",
            "Approx 40% complete for ENCFF349YSG.fastq\n",
            "Approx 45% complete for ENCFF349YSG.fastq\n",
            "Approx 50% complete for ENCFF349YSG.fastq\n",
            "Approx 55% complete for ENCFF349YSG.fastq\n",
            "Approx 60% complete for ENCFF349YSG.fastq\n",
            "Approx 65% complete for ENCFF349YSG.fastq\n",
            "Approx 70% complete for ENCFF349YSG.fastq\n",
            "Approx 75% complete for ENCFF349YSG.fastq\n",
            "Approx 80% complete for ENCFF349YSG.fastq\n",
            "Approx 85% complete for ENCFF349YSG.fastq\n",
            "Approx 90% complete for ENCFF349YSG.fastq\n",
            "Approx 95% complete for ENCFF349YSG.fastq\n",
            "Analysis complete for ENCFF349YSG.fastq\n"
          ]
        }
      ],
      "source": [
        "!./FastQC/fastqc /content/ENCFF815VLD.fastq\n",
        "!./FastQC/fastqc /content/ENCFF644SSR.fastq\n",
        "!./FastQC/fastqc /content/ENCFF349YSG.fastq"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "-TO3DQaUoGhB"
      },
      "source": [
        "Ð”Ð°Ð»ÐµÐµ Ð½ÐµÐ¾Ð±Ñ…Ð¾Ð´Ð¸Ð¼Ð¾ Ð¿Ñ€Ð¾Ð°Ð½Ð°Ð»Ð¸Ð·Ð¸Ñ€Ð¾Ð²Ð°Ñ‚ÑŒ Ð²Ñ‹Ð´Ð°Ñ‡Ñƒ fastqc, Ð¿Ñ€Ð¾Ð¸Ð·Ð²ÐµÑÑ‚Ð¸ Ñ„Ð¸Ð»ÑŒÑ‚Ñ€Ð°Ñ†Ð¸ÑŽ Ð¸Ð»Ð¸ Ð¿Ð¾Ð´Ñ€ÐµÐ·Ð°Ð½Ð¸Ðµ Ñ‡Ñ‚ÐµÐ½Ð¸Ð¹ Ð² ÑÐ»ÑƒÑ‡Ð°Ðµ Ð½ÐµÐ¾Ð±Ñ…Ð¾Ð´Ð¸Ð¼Ð¾ÑÑ‚Ð¸. Ð”Ð»Ñ Ð¿Ñ€Ð¸Ð¼ÐµÑ€Ð° ÑƒÐºÐ°Ð·Ð°Ð½Ð¾ Ð¸ÑÐ¿Ð¾Ð»ÑŒÐ·Ð¾Ð²Ð°Ð½Ð¸Ðµ trimmomatic, Ñ‚Ð°ÐºÐ¶Ðµ Ð¼Ð¾Ð¶Ð½Ð¾ Ð¸ÑÐ¿Ð¾Ð»ÑŒÐ·Ð¾Ð²Ð°Ñ‚ÑŒ seqtk."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 20,
      "metadata": {
        "id": "Qellza2PI7fT",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "77e5dca2-8f2d-4fdb-d4e0-9c6c5bbbf427"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "TrimmomaticSE: Started with arguments:\n",
            " -phred33 ENCFF815VLD.fastq ENCFF815VLD_trimmed.fastq ILLUMINACLIP:TruSeq3-SE:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36\n",
            "Automatically using 2 threads\n",
            "java.io.FileNotFoundException: /content/TruSeq3-SE (No such file or directory)\n",
            "\tat java.base/java.io.FileInputStream.open0(Native Method)\n",
            "\tat java.base/java.io.FileInputStream.open(FileInputStream.java:216)\n",
            "\tat java.base/java.io.FileInputStream.<init>(FileInputStream.java:157)\n",
            "\tat org.usadellab.trimmomatic.fasta.FastaParser.parse(FastaParser.java:54)\n",
            "\tat org.usadellab.trimmomatic.trim.IlluminaClippingTrimmer.loadSequences(IlluminaClippingTrimmer.java:110)\n",
            "\tat org.usadellab.trimmomatic.trim.IlluminaClippingTrimmer.makeIlluminaClippingTrimmer(IlluminaClippingTrimmer.java:71)\n",
            "\tat org.usadellab.trimmomatic.trim.TrimmerFactory.makeTrimmer(TrimmerFactory.java:32)\n",
            "\tat org.usadellab.trimmomatic.Trimmomatic.createTrimmers(Trimmomatic.java:59)\n",
            "\tat org.usadellab.trimmomatic.TrimmomaticSE.run(TrimmomaticSE.java:318)\n",
            "\tat org.usadellab.trimmomatic.Trimmomatic.main(Trimmomatic.java:85)\n",
            "Input Reads: 37425763 Surviving: 31308094 (83.65%) Dropped: 6117669 (16.35%)\n",
            "TrimmomaticSE: Completed successfully\n"
          ]
        }
      ],
      "source": [
        "!trimmomatic SE -phred33 ENCFF815VLD.fastq ENCFF815VLD_trimmed.fastq ILLUMINACLIP:TruSeq3-SE:2:30:10 \\\n",
        "  LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 21,
      "metadata": {
        "id": "PWBBF50gI7mI",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "50bbf2fa-991a-4b89-d73d-9843d60c3755"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Started analysis of ENCFF815VLD_trimmed.fastq\n",
            "Approx 5% complete for ENCFF815VLD_trimmed.fastq\n",
            "Approx 10% complete for ENCFF815VLD_trimmed.fastq\n",
            "Approx 15% complete for ENCFF815VLD_trimmed.fastq\n",
            "Approx 20% complete for ENCFF815VLD_trimmed.fastq\n",
            "Approx 25% complete for ENCFF815VLD_trimmed.fastq\n",
            "Approx 30% complete for ENCFF815VLD_trimmed.fastq\n",
            "Approx 35% complete for ENCFF815VLD_trimmed.fastq\n",
            "Approx 40% complete for ENCFF815VLD_trimmed.fastq\n",
            "Approx 45% complete for ENCFF815VLD_trimmed.fastq\n",
            "Approx 50% complete for ENCFF815VLD_trimmed.fastq\n",
            "Approx 55% complete for ENCFF815VLD_trimmed.fastq\n",
            "Approx 60% complete for ENCFF815VLD_trimmed.fastq\n",
            "Approx 65% complete for ENCFF815VLD_trimmed.fastq\n",
            "Approx 70% complete for ENCFF815VLD_trimmed.fastq\n",
            "Approx 75% complete for ENCFF815VLD_trimmed.fastq\n",
            "Approx 80% complete for ENCFF815VLD_trimmed.fastq\n",
            "Approx 85% complete for ENCFF815VLD_trimmed.fastq\n",
            "Approx 90% complete for ENCFF815VLD_trimmed.fastq\n",
            "Approx 95% complete for ENCFF815VLD_trimmed.fastq\n",
            "Analysis complete for ENCFF815VLD_trimmed.fastq\n"
          ]
        }
      ],
      "source": [
        "!./FastQC/fastqc ENCFF815VLD_trimmed.fastq"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!trimmomatic SE -phred33 ENCFF644SSR.fastq ENCFF644SSR_trimmed.fastq ILLUMINACLIP:TruSeq3-SE:2:30:10 \\\n",
        "  LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36"
      ],
      "metadata": {
        "id": "MZCkYL_VLJo9",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "ffc0dcf6-fec9-4bae-fdca-8e635e1cf343"
      },
      "execution_count": 22,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "TrimmomaticSE: Started with arguments:\n",
            " -phred33 ENCFF644SSR.fastq ENCFF644SSR_trimmed.fastq ILLUMINACLIP:TruSeq3-SE:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36\n",
            "Automatically using 2 threads\n",
            "java.io.FileNotFoundException: /content/TruSeq3-SE (No such file or directory)\n",
            "\tat java.base/java.io.FileInputStream.open0(Native Method)\n",
            "\tat java.base/java.io.FileInputStream.open(FileInputStream.java:216)\n",
            "\tat java.base/java.io.FileInputStream.<init>(FileInputStream.java:157)\n",
            "\tat org.usadellab.trimmomatic.fasta.FastaParser.parse(FastaParser.java:54)\n",
            "\tat org.usadellab.trimmomatic.trim.IlluminaClippingTrimmer.loadSequences(IlluminaClippingTrimmer.java:110)\n",
            "\tat org.usadellab.trimmomatic.trim.IlluminaClippingTrimmer.makeIlluminaClippingTrimmer(IlluminaClippingTrimmer.java:71)\n",
            "\tat org.usadellab.trimmomatic.trim.TrimmerFactory.makeTrimmer(TrimmerFactory.java:32)\n",
            "\tat org.usadellab.trimmomatic.Trimmomatic.createTrimmers(Trimmomatic.java:59)\n",
            "\tat org.usadellab.trimmomatic.TrimmomaticSE.run(TrimmomaticSE.java:318)\n",
            "\tat org.usadellab.trimmomatic.Trimmomatic.main(Trimmomatic.java:85)\n",
            "Input Reads: 66685589 Surviving: 63038977 (94.53%) Dropped: 3646612 (5.47%)\n",
            "TrimmomaticSE: Completed successfully\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!./FastQC/fastqc ENCFF644SSR_trimmed.fastq"
      ],
      "metadata": {
        "id": "cMX-8zYuLJwn",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "f635fa06-466a-4a5b-ef37-acdbf77e65b3"
      },
      "execution_count": 23,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Started analysis of ENCFF644SSR_trimmed.fastq\n",
            "Approx 5% complete for ENCFF644SSR_trimmed.fastq\n",
            "Approx 10% complete for ENCFF644SSR_trimmed.fastq\n",
            "Approx 15% complete for ENCFF644SSR_trimmed.fastq\n",
            "Approx 20% complete for ENCFF644SSR_trimmed.fastq\n",
            "Approx 25% complete for ENCFF644SSR_trimmed.fastq\n",
            "Approx 30% complete for ENCFF644SSR_trimmed.fastq\n",
            "Approx 35% complete for ENCFF644SSR_trimmed.fastq\n",
            "Approx 40% complete for ENCFF644SSR_trimmed.fastq\n",
            "Approx 45% complete for ENCFF644SSR_trimmed.fastq\n",
            "Approx 50% complete for ENCFF644SSR_trimmed.fastq\n",
            "Approx 55% complete for ENCFF644SSR_trimmed.fastq\n",
            "Approx 60% complete for ENCFF644SSR_trimmed.fastq\n",
            "Approx 65% complete for ENCFF644SSR_trimmed.fastq\n",
            "Approx 70% complete for ENCFF644SSR_trimmed.fastq\n",
            "Approx 75% complete for ENCFF644SSR_trimmed.fastq\n",
            "Approx 80% complete for ENCFF644SSR_trimmed.fastq\n",
            "Approx 85% complete for ENCFF644SSR_trimmed.fastq\n",
            "Approx 90% complete for ENCFF644SSR_trimmed.fastq\n",
            "Approx 95% complete for ENCFF644SSR_trimmed.fastq\n",
            "Analysis complete for ENCFF644SSR_trimmed.fastq\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!trimmomatic SE -phred33 ENCFF349YSG.fastq ENCFF349YSG_trimmed.fastq ILLUMINACLIP:TruSeq3-SE:2:30:10 \\\n",
        "  LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36"
      ],
      "metadata": {
        "id": "ZDA_pVAMLJ47",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "4f345dca-bf38-4f19-b2cc-1bdb89b182c5"
      },
      "execution_count": 24,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "TrimmomaticSE: Started with arguments:\n",
            " -phred33 ENCFF349YSG.fastq ENCFF349YSG_trimmed.fastq ILLUMINACLIP:TruSeq3-SE:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36\n",
            "Automatically using 2 threads\n",
            "java.io.FileNotFoundException: /content/TruSeq3-SE (No such file or directory)\n",
            "\tat java.base/java.io.FileInputStream.open0(Native Method)\n",
            "\tat java.base/java.io.FileInputStream.open(FileInputStream.java:216)\n",
            "\tat java.base/java.io.FileInputStream.<init>(FileInputStream.java:157)\n",
            "\tat org.usadellab.trimmomatic.fasta.FastaParser.parse(FastaParser.java:54)\n",
            "\tat org.usadellab.trimmomatic.trim.IlluminaClippingTrimmer.loadSequences(IlluminaClippingTrimmer.java:110)\n",
            "\tat org.usadellab.trimmomatic.trim.IlluminaClippingTrimmer.makeIlluminaClippingTrimmer(IlluminaClippingTrimmer.java:71)\n",
            "\tat org.usadellab.trimmomatic.trim.TrimmerFactory.makeTrimmer(TrimmerFactory.java:32)\n",
            "\tat org.usadellab.trimmomatic.Trimmomatic.createTrimmers(Trimmomatic.java:59)\n",
            "\tat org.usadellab.trimmomatic.TrimmomaticSE.run(TrimmomaticSE.java:318)\n",
            "\tat org.usadellab.trimmomatic.Trimmomatic.main(Trimmomatic.java:85)\n",
            "Input Reads: 10167581 Surviving: 9808524 (96.47%) Dropped: 359057 (3.53%)\n",
            "TrimmomaticSE: Completed successfully\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!./FastQC/fastqc ENCFF349YSG_trimmed.fastq"
      ],
      "metadata": {
        "id": "HhlMPXy-LKAA",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "36cdf1fd-56b4-4ce0-f5e4-6cc400822b4a"
      },
      "execution_count": 25,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Started analysis of ENCFF349YSG_trimmed.fastq\n",
            "Approx 5% complete for ENCFF349YSG_trimmed.fastq\n",
            "Approx 10% complete for ENCFF349YSG_trimmed.fastq\n",
            "Approx 15% complete for ENCFF349YSG_trimmed.fastq\n",
            "Approx 20% complete for ENCFF349YSG_trimmed.fastq\n",
            "Approx 25% complete for ENCFF349YSG_trimmed.fastq\n",
            "Approx 30% complete for ENCFF349YSG_trimmed.fastq\n",
            "Approx 35% complete for ENCFF349YSG_trimmed.fastq\n",
            "Approx 40% complete for ENCFF349YSG_trimmed.fastq\n",
            "Approx 45% complete for ENCFF349YSG_trimmed.fastq\n",
            "Approx 50% complete for ENCFF349YSG_trimmed.fastq\n",
            "Approx 55% complete for ENCFF349YSG_trimmed.fastq\n",
            "Approx 60% complete for ENCFF349YSG_trimmed.fastq\n",
            "Approx 65% complete for ENCFF349YSG_trimmed.fastq\n",
            "Approx 70% complete for ENCFF349YSG_trimmed.fastq\n",
            "Approx 75% complete for ENCFF349YSG_trimmed.fastq\n",
            "Approx 80% complete for ENCFF349YSG_trimmed.fastq\n",
            "Approx 85% complete for ENCFF349YSG_trimmed.fastq\n",
            "Approx 90% complete for ENCFF349YSG_trimmed.fastq\n",
            "Approx 95% complete for ENCFF349YSG_trimmed.fastq\n",
            "Analysis complete for ENCFF349YSG_trimmed.fastq\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "W9QHWBLlpMjQ"
      },
      "source": [
        "## Ð’Ñ‹Ñ€Ð°Ð²Ð½Ð¸Ð²Ð°Ð½Ð¸Ðµ Ð½Ð° Ñ…Ñ€Ð¾Ð¼Ð¾ÑÐ¾Ð¼Ñƒ\n",
        "\n",
        "Ð¡Ð»ÐµÐ´ÑƒÐµÑ‚ Ð²Ñ‹Ð±Ñ€Ð°Ñ‚ÑŒ Ð¾Ð´Ð½Ñƒ Ñ…Ñ€Ð¾Ð¼Ð¾ÑÐ¾Ð¼Ñƒ, Ð¿Ð¾Ñ‚Ð¾Ð¼Ñƒ Ñ‡Ñ‚Ð¾ Ñ€ÐµÑÑƒÑ€ÑÑ‹ google colab Ð¾Ð³Ñ€Ð°Ð½Ð¸Ñ‡ÐµÐ½Ñ‹. Ð’ ÑÐ»ÑƒÑ‡Ð°Ðµ Ð¿Ñ€Ð¾Ð±Ð»ÐµÐ¼, Ñ‚Ð°ÐºÐ¸Ñ… ÐºÐ°Ðº Segmentation fault, error 139, ÑÐ»ÐµÐ´ÑƒÐµÑ‚ Ð²Ñ‹Ð±Ñ€Ð°Ñ‚ÑŒ Ñ…Ñ€Ð¾Ð¼Ð¾ÑÐ¾Ð¼Ñƒ Ð¼ÐµÐ½ÑŒÑˆÐµÐ³Ð¾ Ñ€Ð°Ð·Ð¼ÐµÑ€Ð°.\n",
        "\n",
        "ÐŸÐ¾ÑÐ»ÐµÐ´Ð¾Ð²Ð°Ñ‚ÐµÐ»ÑŒÐ½Ð¾ÑÑ‚ÑŒ Ð½ÑƒÐºÐ»ÐµÐ¾Ñ‚Ð¸Ð´Ð¾Ð² Ñ…Ñ€Ð¾Ð¼Ð¾ÑÐ¾Ð¼Ñ‹ Ð¼Ð¾Ð¶Ð½Ð¾ ÑÐºÐ°Ñ‡Ð°Ñ‚ÑŒ Ð¿Ð¾ Ð°Ð´Ñ€ÐµÑÑƒ https://hgdownload.soe.ucsc.edu/goldenPath/hg38/chromosomes/"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 26,
      "metadata": {
        "id": "N4JIGZOsGiY2",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "b5a5c99f-fd40-4a2e-c1d2-e5eeb0be5fbb"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2023-03-01 18:39:44--  https://hgdownload.soe.ucsc.edu/goldenPath/hg38/chromosomes/chr16.fa.gz\n",
            "Resolving hgdownload.soe.ucsc.edu (hgdownload.soe.ucsc.edu)... 128.114.119.163\n",
            "Connecting to hgdownload.soe.ucsc.edu (hgdownload.soe.ucsc.edu)|128.114.119.163|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 26106261 (25M) [application/x-gzip]\n",
            "Saving to: â€˜chr16.fa.gzâ€™\n",
            "\n",
            "chr16.fa.gz         100%[===================>]  24.90M  74.9MB/s    in 0.3s    \n",
            "\n",
            "2023-03-01 18:39:45 (74.9 MB/s) - â€˜chr16.fa.gzâ€™ saved [26106261/26106261]\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!wget https://hgdownload.soe.ucsc.edu/goldenPath/hg38/chromosomes/chr16.fa.gz\n",
        "!gzip -d chr16.fa.gz"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 27,
      "metadata": {
        "id": "bhraQ6NdGiPN",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "814478b3-874d-4286-fb47-02ae31689636"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Settings:\n",
            "  Output files: \"chromosome_index.*.bt2\"\n",
            "  Line rate: 6 (line is 64 bytes)\n",
            "  Lines per side: 1 (side is 64 bytes)\n",
            "  Offset rate: 4 (one in 16)\n",
            "  FTable chars: 10\n",
            "  Strings: unpacked\n",
            "  Max bucket size: default\n",
            "  Max bucket size, sqrt multiplier: default\n",
            "  Max bucket size, len divisor: 4\n",
            "  Difference-cover sample period: 1024\n",
            "  Endianness: little\n",
            "  Actual local endianness: little\n",
            "  Sanity checking: disabled\n",
            "  Assertions: disabled\n",
            "  Random seed: 0\n",
            "  Sizeofs: void*:8, int:4, long:8, size_t:8\n",
            "Input files DNA, FASTA:\n",
            "  chr16.fa\n",
            "Building a SMALL index\n",
            "Reading reference sizes\n",
            "  Time reading reference sizes: 00:00:01\n",
            "Calculating joined length\n",
            "Writing header\n",
            "Reserving space for joined string\n",
            "Joining reference sequences\n",
            "  Time to join reference sequences: 00:00:01\n",
            "bmax according to bmaxDivN setting: 20451485\n",
            "Using parameters --bmax 15338614 --dcv 1024\n",
            "  Doing ahead-of-time memory usage test\n",
            "  Passed!  Constructing with these parameters: --bmax 15338614 --dcv 1024\n",
            "Constructing suffix-array element generator\n",
            "Building DifferenceCoverSample\n",
            "  Building sPrime\n",
            "  Building sPrimeOrder\n",
            "  V-Sorting samples\n",
            "  V-Sorting samples time: 00:00:03\n",
            "  Allocating rank array\n",
            "  Ranking v-sort output\n",
            "  Ranking v-sort output time: 00:00:01\n",
            "  Invoking Larsson-Sadakane on ranks\n",
            "  Invoking Larsson-Sadakane on ranks time: 00:00:00\n",
            "  Sanity-checking and returning\n",
            "Building samples\n",
            "Reserving space for 12 sample suffixes\n",
            "Generating random suffixes\n",
            "QSorting 12 sample offsets, eliminating duplicates\n",
            "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
            "Multikey QSorting 12 samples\n",
            "  (Using difference cover)\n",
            "  Multikey QSorting samples time: 00:00:00\n",
            "Calculating bucket sizes\n",
            "Splitting and merging\n",
            "  Splitting and merging time: 00:00:00\n",
            "Avg bucket size: 1.16866e+07 (target: 15338613)\n",
            "Converting suffix-array elements to index image\n",
            "Allocating ftab, absorbFtab\n",
            "Entering Ebwt loop\n",
            "Getting block 1 of 7\n",
            "  Reserving size (15338614) for bucket 1\n",
            "  Calculating Z arrays for bucket 1\n",
            "  Entering block accumulator loop for bucket 1:\n",
            "  bucket 1: 10%\n",
            "  bucket 1: 20%\n",
            "  bucket 1: 30%\n",
            "  bucket 1: 40%\n",
            "  bucket 1: 50%\n",
            "  bucket 1: 60%\n",
            "  bucket 1: 70%\n",
            "  bucket 1: 80%\n",
            "  bucket 1: 90%\n",
            "  bucket 1: 100%\n",
            "  Sorting block of length 14716821 for bucket 1\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:12\n",
            "Returning block of 14716822 for bucket 1\n",
            "Getting block 2 of 7\n",
            "  Reserving size (15338614) for bucket 2\n",
            "  Calculating Z arrays for bucket 2\n",
            "  Entering block accumulator loop for bucket 2:\n",
            "  bucket 2: 10%\n",
            "  bucket 2: 20%\n",
            "  bucket 2: 30%\n",
            "  bucket 2: 40%\n",
            "  bucket 2: 50%\n",
            "  bucket 2: 60%\n",
            "  bucket 2: 70%\n",
            "  bucket 2: 80%\n",
            "  bucket 2: 90%\n",
            "  bucket 2: 100%\n",
            "  Sorting block of length 7564940 for bucket 2\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:05\n",
            "Returning block of 7564941 for bucket 2\n",
            "Getting block 3 of 7\n",
            "  Reserving size (15338614) for bucket 3\n",
            "  Calculating Z arrays for bucket 3\n",
            "  Entering block accumulator loop for bucket 3:\n",
            "  bucket 3: 10%\n",
            "  bucket 3: 20%\n",
            "  bucket 3: 30%\n",
            "  bucket 3: 40%\n",
            "  bucket 3: 50%\n",
            "  bucket 3: 60%\n",
            "  bucket 3: 70%\n",
            "  bucket 3: 80%\n",
            "  bucket 3: 90%\n",
            "  bucket 3: 100%\n",
            "  Sorting block of length 12102407 for bucket 3\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:12\n",
            "Returning block of 12102408 for bucket 3\n",
            "Getting block 4 of 7\n",
            "  Reserving size (15338614) for bucket 4\n",
            "  Calculating Z arrays for bucket 4\n",
            "  Entering block accumulator loop for bucket 4:\n",
            "  bucket 4: 10%\n",
            "  bucket 4: 20%\n",
            "  bucket 4: 30%\n",
            "  bucket 4: 40%\n",
            "  bucket 4: 50%\n",
            "  bucket 4: 60%\n",
            "  bucket 4: 70%\n",
            "  bucket 4: 80%\n",
            "  bucket 4: 90%\n",
            "  bucket 4: 100%\n",
            "  Sorting block of length 14907022 for bucket 4\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:12\n",
            "Returning block of 14907023 for bucket 4\n",
            "Getting block 5 of 7\n",
            "  Reserving size (15338614) for bucket 5\n",
            "  Calculating Z arrays for bucket 5\n",
            "  Entering block accumulator loop for bucket 5:\n",
            "  bucket 5: 10%\n",
            "  bucket 5: 20%\n",
            "  bucket 5: 30%\n",
            "  bucket 5: 40%\n",
            "  bucket 5: 50%\n",
            "  bucket 5: 60%\n",
            "  bucket 5: 70%\n",
            "  bucket 5: 80%\n",
            "  bucket 5: 90%\n",
            "  bucket 5: 100%\n",
            "  Sorting block of length 3528554 for bucket 5\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:03\n",
            "Returning block of 3528555 for bucket 5\n",
            "Getting block 6 of 7\n",
            "  Reserving size (15338614) for bucket 6\n",
            "  Calculating Z arrays for bucket 6\n",
            "  Entering block accumulator loop for bucket 6:\n",
            "  bucket 6: 10%\n",
            "  bucket 6: 20%\n",
            "  bucket 6: 30%\n",
            "  bucket 6: 40%\n",
            "  bucket 6: 50%\n",
            "  bucket 6: 60%\n",
            "  bucket 6: 70%\n",
            "  bucket 6: 80%\n",
            "  bucket 6: 90%\n",
            "  bucket 6: 100%\n",
            "  Sorting block of length 14550274 for bucket 6\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:12\n",
            "Returning block of 14550275 for bucket 6\n",
            "Getting block 7 of 7\n",
            "  Reserving size (15338614) for bucket 7\n",
            "  Calculating Z arrays for bucket 7\n",
            "  Entering block accumulator loop for bucket 7:\n",
            "  bucket 7: 10%\n",
            "  bucket 7: 20%\n",
            "  bucket 7: 30%\n",
            "  bucket 7: 40%\n",
            "  bucket 7: 50%\n",
            "  bucket 7: 60%\n",
            "  bucket 7: 70%\n",
            "  bucket 7: 80%\n",
            "  bucket 7: 90%\n",
            "  bucket 7: 100%\n",
            "  Sorting block of length 14435919 for bucket 7\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:10\n",
            "Returning block of 14435920 for bucket 7\n",
            "Exited Ebwt loop\n",
            "fchr[A]: 0\n",
            "fchr[C]: 22558319\n",
            "fchr[G]: 40731061\n",
            "fchr[T]: 59031037\n",
            "fchr[$]: 81805943\n",
            "Exiting Ebwt::buildToDisk()\n",
            "Returning from initFromVector\n",
            "Wrote 31463383 bytes to primary EBWT file: chromosome_index.1.bt2.tmp\n",
            "Wrote 20451492 bytes to secondary EBWT file: chromosome_index.2.bt2.tmp\n",
            "Re-opening _in1 and _in2 as input streams\n",
            "Returning from Ebwt constructor\n",
            "Headers:\n",
            "    len: 81805943\n",
            "    bwtLen: 81805944\n",
            "    sz: 20451486\n",
            "    bwtSz: 20451486\n",
            "    lineRate: 6\n",
            "    offRate: 4\n",
            "    offMask: 0xfffffff0\n",
            "    ftabChars: 10\n",
            "    eftabLen: 20\n",
            "    eftabSz: 80\n",
            "    ftabLen: 1048577\n",
            "    ftabSz: 4194308\n",
            "    offsLen: 5112872\n",
            "    offsSz: 20451488\n",
            "    lineSz: 64\n",
            "    sideSz: 64\n",
            "    sideBwtSz: 48\n",
            "    sideBwtLen: 192\n",
            "    numSides: 426073\n",
            "    numLines: 426073\n",
            "    ebwtTotLen: 27268672\n",
            "    ebwtTotSz: 27268672\n",
            "    color: 0\n",
            "    reverse: 0\n",
            "Total time for call to driver() for forward index: 00:01:53\n",
            "Reading reference sizes\n",
            "  Time reading reference sizes: 00:00:01\n",
            "Calculating joined length\n",
            "Writing header\n",
            "Reserving space for joined string\n",
            "Joining reference sequences\n",
            "  Time to join reference sequences: 00:00:01\n",
            "  Time to reverse reference sequence: 00:00:00\n",
            "bmax according to bmaxDivN setting: 20451485\n",
            "Using parameters --bmax 15338614 --dcv 1024\n",
            "  Doing ahead-of-time memory usage test\n",
            "  Passed!  Constructing with these parameters: --bmax 15338614 --dcv 1024\n",
            "Constructing suffix-array element generator\n",
            "Building DifferenceCoverSample\n",
            "  Building sPrime\n",
            "  Building sPrimeOrder\n",
            "  V-Sorting samples\n",
            "  V-Sorting samples time: 00:00:04\n",
            "  Allocating rank array\n",
            "  Ranking v-sort output\n",
            "  Ranking v-sort output time: 00:00:00\n",
            "  Invoking Larsson-Sadakane on ranks\n",
            "  Invoking Larsson-Sadakane on ranks time: 00:00:01\n",
            "  Sanity-checking and returning\n",
            "Building samples\n",
            "Reserving space for 12 sample suffixes\n",
            "Generating random suffixes\n",
            "QSorting 12 sample offsets, eliminating duplicates\n",
            "QSorting sample offsets, eliminating duplicates time: 00:00:00\n",
            "Multikey QSorting 12 samples\n",
            "  (Using difference cover)\n",
            "  Multikey QSorting samples time: 00:00:00\n",
            "Calculating bucket sizes\n",
            "Splitting and merging\n",
            "  Splitting and merging time: 00:00:00\n",
            "Split 1, merged 6; iterating...\n",
            "Splitting and merging\n",
            "  Splitting and merging time: 00:00:00\n",
            "Avg bucket size: 1.16866e+07 (target: 15338613)\n",
            "Converting suffix-array elements to index image\n",
            "Allocating ftab, absorbFtab\n",
            "Entering Ebwt loop\n",
            "Getting block 1 of 7\n",
            "  Reserving size (15338614) for bucket 1\n",
            "  Calculating Z arrays for bucket 1\n",
            "  Entering block accumulator loop for bucket 1:\n",
            "  bucket 1: 10%\n",
            "  bucket 1: 20%\n",
            "  bucket 1: 30%\n",
            "  bucket 1: 40%\n",
            "  bucket 1: 50%\n",
            "  bucket 1: 60%\n",
            "  bucket 1: 70%\n",
            "  bucket 1: 80%\n",
            "  bucket 1: 90%\n",
            "  bucket 1: 100%\n",
            "  Sorting block of length 13967169 for bucket 1\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:12\n",
            "Returning block of 13967170 for bucket 1\n",
            "Getting block 2 of 7\n",
            "  Reserving size (15338614) for bucket 2\n",
            "  Calculating Z arrays for bucket 2\n",
            "  Entering block accumulator loop for bucket 2:\n",
            "  bucket 2: 10%\n",
            "  bucket 2: 20%\n",
            "  bucket 2: 30%\n",
            "  bucket 2: 40%\n",
            "  bucket 2: 50%\n",
            "  bucket 2: 60%\n",
            "  bucket 2: 70%\n",
            "  bucket 2: 80%\n",
            "  bucket 2: 90%\n",
            "  bucket 2: 100%\n",
            "  Sorting block of length 8910540 for bucket 2\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:06\n",
            "Returning block of 8910541 for bucket 2\n",
            "Getting block 3 of 7\n",
            "  Reserving size (15338614) for bucket 3\n",
            "  Calculating Z arrays for bucket 3\n",
            "  Entering block accumulator loop for bucket 3:\n",
            "  bucket 3: 10%\n",
            "  bucket 3: 20%\n",
            "  bucket 3: 30%\n",
            "  bucket 3: 40%\n",
            "  bucket 3: 50%\n",
            "  bucket 3: 60%\n",
            "  bucket 3: 70%\n",
            "  bucket 3: 80%\n",
            "  bucket 3: 90%\n",
            "  bucket 3: 100%\n",
            "  Sorting block of length 11630282 for bucket 3\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:09\n",
            "Returning block of 11630283 for bucket 3\n",
            "Getting block 4 of 7\n",
            "  Reserving size (15338614) for bucket 4\n",
            "  Calculating Z arrays for bucket 4\n",
            "  Entering block accumulator loop for bucket 4:\n",
            "  bucket 4: 10%\n",
            "  bucket 4: 20%\n",
            "  bucket 4: 30%\n",
            "  bucket 4: 40%\n",
            "  bucket 4: 50%\n",
            "  bucket 4: 60%\n",
            "  bucket 4: 70%\n",
            "  bucket 4: 80%\n",
            "  bucket 4: 90%\n",
            "  bucket 4: 100%\n",
            "  Sorting block of length 13598916 for bucket 4\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:12\n",
            "Returning block of 13598917 for bucket 4\n",
            "Getting block 5 of 7\n",
            "  Reserving size (15338614) for bucket 5\n",
            "  Calculating Z arrays for bucket 5\n",
            "  Entering block accumulator loop for bucket 5:\n",
            "  bucket 5: 10%\n",
            "  bucket 5: 20%\n",
            "  bucket 5: 30%\n",
            "  bucket 5: 40%\n",
            "  bucket 5: 50%\n",
            "  bucket 5: 60%\n",
            "  bucket 5: 70%\n",
            "  bucket 5: 80%\n",
            "  bucket 5: 90%\n",
            "  bucket 5: 100%\n",
            "  Sorting block of length 10898776 for bucket 5\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:07\n",
            "Returning block of 10898777 for bucket 5\n",
            "Getting block 6 of 7\n",
            "  Reserving size (15338614) for bucket 6\n",
            "  Calculating Z arrays for bucket 6\n",
            "  Entering block accumulator loop for bucket 6:\n",
            "  bucket 6: 10%\n",
            "  bucket 6: 20%\n",
            "  bucket 6: 30%\n",
            "  bucket 6: 40%\n",
            "  bucket 6: 50%\n",
            "  bucket 6: 60%\n",
            "  bucket 6: 70%\n",
            "  bucket 6: 80%\n",
            "  bucket 6: 90%\n",
            "  bucket 6: 100%\n",
            "  Sorting block of length 12685008 for bucket 6\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:10\n",
            "Returning block of 12685009 for bucket 6\n",
            "Getting block 7 of 7\n",
            "  Reserving size (15338614) for bucket 7\n",
            "  Calculating Z arrays for bucket 7\n",
            "  Entering block accumulator loop for bucket 7:\n",
            "  bucket 7: 10%\n",
            "  bucket 7: 20%\n",
            "  bucket 7: 30%\n",
            "  bucket 7: 40%\n",
            "  bucket 7: 50%\n",
            "  bucket 7: 60%\n",
            "  bucket 7: 70%\n",
            "  bucket 7: 80%\n",
            "  bucket 7: 90%\n",
            "  bucket 7: 100%\n",
            "  Sorting block of length 10115246 for bucket 7\n",
            "  (Using difference cover)\n",
            "  Sorting block time: 00:00:08\n",
            "Returning block of 10115247 for bucket 7\n",
            "Exited Ebwt loop\n",
            "fchr[A]: 0\n",
            "fchr[C]: 22558319\n",
            "fchr[G]: 40731061\n",
            "fchr[T]: 59031037\n",
            "fchr[$]: 81805943\n",
            "Exiting Ebwt::buildToDisk()\n",
            "Returning from initFromVector\n",
            "Wrote 31463383 bytes to primary EBWT file: chromosome_index.rev.1.bt2.tmp\n",
            "Wrote 20451492 bytes to secondary EBWT file: chromosome_index.rev.2.bt2.tmp\n",
            "Re-opening _in1 and _in2 as input streams\n",
            "Returning from Ebwt constructor\n",
            "Headers:\n",
            "    len: 81805943\n",
            "    bwtLen: 81805944\n",
            "    sz: 20451486\n",
            "    bwtSz: 20451486\n",
            "    lineRate: 6\n",
            "    offRate: 4\n",
            "    offMask: 0xfffffff0\n",
            "    ftabChars: 10\n",
            "    eftabLen: 20\n",
            "    eftabSz: 80\n",
            "    ftabLen: 1048577\n",
            "    ftabSz: 4194308\n",
            "    offsLen: 5112872\n",
            "    offsSz: 20451488\n",
            "    lineSz: 64\n",
            "    sideSz: 64\n",
            "    sideBwtSz: 48\n",
            "    sideBwtLen: 192\n",
            "    numSides: 426073\n",
            "    numLines: 426073\n",
            "    ebwtTotLen: 27268672\n",
            "    ebwtTotSz: 27268672\n",
            "    color: 0\n",
            "    reverse: 1\n",
            "Total time for backward call to driver() for mirror index: 00:01:50\n",
            "Renaming chromosome_index.3.bt2.tmp to chromosome_index.3.bt2\n",
            "Renaming chromosome_index.4.bt2.tmp to chromosome_index.4.bt2\n",
            "Renaming chromosome_index.1.bt2.tmp to chromosome_index.1.bt2\n",
            "Renaming chromosome_index.2.bt2.tmp to chromosome_index.2.bt2\n",
            "Renaming chromosome_index.rev.1.bt2.tmp to chromosome_index.rev.1.bt2\n",
            "Renaming chromosome_index.rev.2.bt2.tmp to chromosome_index.rev.2.bt2\n",
            "CPU times: user 2.43 s, sys: 245 ms, total: 2.68 s\n",
            "Wall time: 3min 43s\n"
          ]
        }
      ],
      "source": [
        "%%time\n",
        "!bowtie2-build chr16.fa chromosome_index"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 28,
      "metadata": {
        "id": "E_usI8XUrQnR"
      },
      "outputs": [],
      "source": [
        "!mkdir bowtie2_res"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 29,
      "metadata": {
        "id": "3t80UATHqFe7",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "1324ba4c-865a-4f5c-cb90-2041b8d95b8c"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "31308094 reads; of these:\n",
            "  31308094 (100.00%) were unpaired; of these:\n",
            "    27145342 (86.70%) aligned 0 times\n",
            "    890736 (2.85%) aligned exactly 1 time\n",
            "    3272016 (10.45%) aligned >1 times\n",
            "13.30% overall alignment rate\n",
            "CPU times: user 12.8 s, sys: 1.17 s, total: 14 s\n",
            "Wall time: 20min 58s\n"
          ]
        }
      ],
      "source": [
        "%%time\n",
        "!bowtie2  -p 2 \\\n",
        "          -x chromosome_index \\\n",
        "          -U ENCFF815VLD_trimmed.fastq \\\n",
        "          -S bowtie2_res/res_VLD.sam"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 30,
      "metadata": {
        "id": "so1jTdGHoEir",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "075a56da-5447-44a6-8d49-1a8087968845"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "63038977 reads; of these:\n",
            "  63038977 (100.00%) were unpaired; of these:\n",
            "    54471818 (86.41%) aligned 0 times\n",
            "    1833175 (2.91%) aligned exactly 1 time\n",
            "    6733984 (10.68%) aligned >1 times\n",
            "13.59% overall alignment rate\n",
            "CPU times: user 25.3 s, sys: 2.17 s, total: 27.4 s\n",
            "Wall time: 42min 5s\n"
          ]
        }
      ],
      "source": [
        "%%time\n",
        "!bowtie2  -p 2 \\\n",
        "          -x chromosome_index \\\n",
        "          -U ENCFF644SSR_trimmed.fastq \\\n",
        "          -S bowtie2_res/res_SSR.sam"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "%%time\n",
        "!bowtie2  -p 2 \\\n",
        "          -x chromosome_index \\\n",
        "          -U ENCFF349YSG_trimmed.fastq \\\n",
        "          -S bowtie2_res/res_YSG.sam"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "GaYwaQZnemoi",
        "outputId": "4128ea9f-2316-41b8-f835-406dd8399412"
      },
      "execution_count": 31,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "9808524 reads; of these:\n",
            "  9808524 (100.00%) were unpaired; of these:\n",
            "    8511280 (86.77%) aligned 0 times\n",
            "    293587 (2.99%) aligned exactly 1 time\n",
            "    1003657 (10.23%) aligned >1 times\n",
            "13.23% overall alignment rate\n",
            "CPU times: user 11.7 s, sys: 1.04 s, total: 12.7 s\n",
            "Wall time: 19min 24s\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "ysoDrnSCrei0"
      },
      "source": [
        "ÐŸÑ€Ð¾Ð°Ð½Ð°Ð»Ð¸Ð·Ð¸Ñ€ÑƒÐ¹Ñ‚Ðµ Ð²Ñ‹Ð´Ð°Ñ‡Ñƒ bowtie. ÐŸÐ¾Ñ‡ÐµÐ¼Ñƒ Ð¿Ñ€Ð¾Ñ†ÐµÐ½Ñ‚ Ð²Ñ‹Ñ€Ð°Ð²Ð½Ð¸Ð²Ð°Ð½Ð¸Ð¹ Ð¿Ð¾Ð»ÑƒÑ‡Ð¸Ð»ÑÑ Ð¸Ð¼ÐµÐ½Ð½Ð¾ Ñ‚Ð°ÐºÐ¸Ð¼?\n",
        "\n",
        "Ð˜Ð¼ÐµÐµÑ‚ ÑÐ¼Ñ‹ÑÐ» Ð´Ð»Ñ Ð´Ð°Ð»ÑŒÐ½ÐµÐ¹ÑˆÐµÐ³Ð¾ Ð°Ð½Ð°Ð»Ð¸Ð·Ð° Ð¾Ñ‚Ð¾Ð±Ñ€Ð°Ñ‚ÑŒ ÑƒÐ½Ð¸ÐºÐ°Ð»ÑŒÐ½Ð¾ ÐºÐ°Ñ€Ñ‚Ð¸Ñ€Ð¾Ð²Ð°Ð½Ð½Ñ‹Ðµ Ñ€Ð¸Ð´Ñ‹."
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "## Ð¢Ð°Ð±Ð»Ð¸Ñ†Ð°\n"
      ],
      "metadata": {
        "id": "IA5lpWAshFrv"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "import pandas as pd\n",
        "data = {'ÐžÐ±Ñ‰ÐµÐµ Ñ‡Ð¸ÑÐ»Ð¾ Ñ€Ð¸Ð´Ð¾Ð²':[31308094, 63038977, 9808524],\n",
        "        'Ð’Ñ‹Ñ€Ð¾Ð²Ð½Ð¸Ð²ÑˆÐ¸ÐµÑÑ 1 Ñ€Ð°Ð·':['890736 (2.85%)', '1833175 (2.91%)', '293587 (2.99%)'],\n",
        "        'Ð’Ñ‹Ñ€Ð¾Ð²Ð½Ð¸Ð²ÑˆÐ¸ÐµÑÑ Ð±Ð¾Ð»ÑŒÑˆÐµ 1 Ñ€Ð°Ð·Ð°':['3272016 (10.45%)', '6733984 (10.68%)', '1003657 (10.23%)'],\n",
        "        'ÐÐµÐ²Ñ‹Ñ€Ð¾Ð²Ð½Ð¸Ð²ÑˆÐ¸ÐµÑÑ':['27145342 (86.70%)', '54471818 (86.41%)', '8511280 (86.77%)']}\n",
        "\n",
        "table = pd.DataFrame(data, index =['ENCFF815VLD', 'ENCFF644SSR', 'ENCFF349YSG'])\n",
        "table"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 143
        },
        "id": "M7U29FEQhQu3",
        "outputId": "215f1642-c4c5-4241-8964-b76d7f23655a"
      },
      "execution_count": 32,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "             ÐžÐ±Ñ‰ÐµÐµ Ñ‡Ð¸ÑÐ»Ð¾ Ñ€Ð¸Ð´Ð¾Ð² Ð’Ñ‹Ñ€Ð¾Ð²Ð½Ð¸Ð²ÑˆÐ¸ÐµÑÑ 1 Ñ€Ð°Ð·  \\\n",
              "ENCFF815VLD           31308094      890736 (2.85%)   \n",
              "ENCFF644SSR           63038977     1833175 (2.91%)   \n",
              "ENCFF349YSG            9808524      293587 (2.99%)   \n",
              "\n",
              "            Ð’Ñ‹Ñ€Ð¾Ð²Ð½Ð¸Ð²ÑˆÐ¸ÐµÑÑ Ð±Ð¾Ð»ÑŒÑˆÐµ 1 Ñ€Ð°Ð·Ð°    ÐÐµÐ²Ñ‹Ñ€Ð¾Ð²Ð½Ð¸Ð²ÑˆÐ¸ÐµÑÑ  \n",
              "ENCFF815VLD            3272016 (10.45%)  27145342 (86.70%)  \n",
              "ENCFF644SSR            6733984 (10.68%)  54471818 (86.41%)  \n",
              "ENCFF349YSG            1003657 (10.23%)   8511280 (86.77%)  "
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-d671c676-4248-4715-be15-c885b3e422f9\">\n",
              "    <div class=\"colab-df-container\">\n",
              "      <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th></th>\n",
              "      <th>ÐžÐ±Ñ‰ÐµÐµ Ñ‡Ð¸ÑÐ»Ð¾ Ñ€Ð¸Ð´Ð¾Ð²</th>\n",
              "      <th>Ð’Ñ‹Ñ€Ð¾Ð²Ð½Ð¸Ð²ÑˆÐ¸ÐµÑÑ 1 Ñ€Ð°Ð·</th>\n",
              "      <th>Ð’Ñ‹Ñ€Ð¾Ð²Ð½Ð¸Ð²ÑˆÐ¸ÐµÑÑ Ð±Ð¾Ð»ÑŒÑˆÐµ 1 Ñ€Ð°Ð·Ð°</th>\n",
              "      <th>ÐÐµÐ²Ñ‹Ñ€Ð¾Ð²Ð½Ð¸Ð²ÑˆÐ¸ÐµÑÑ</th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>ENCFF815VLD</th>\n",
              "      <td>31308094</td>\n",
              "      <td>890736 (2.85%)</td>\n",
              "      <td>3272016 (10.45%)</td>\n",
              "      <td>27145342 (86.70%)</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>ENCFF644SSR</th>\n",
              "      <td>63038977</td>\n",
              "      <td>1833175 (2.91%)</td>\n",
              "      <td>6733984 (10.68%)</td>\n",
              "      <td>54471818 (86.41%)</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>ENCFF349YSG</th>\n",
              "      <td>9808524</td>\n",
              "      <td>293587 (2.99%)</td>\n",
              "      <td>1003657 (10.23%)</td>\n",
              "      <td>8511280 (86.77%)</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "      <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-d671c676-4248-4715-be15-c885b3e422f9')\"\n",
              "              title=\"Convert this dataframe to an interactive table.\"\n",
              "              style=\"display:none;\">\n",
              "        \n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "       width=\"24px\">\n",
              "    <path d=\"M0 0h24v24H0V0z\" fill=\"none\"/>\n",
              "    <path d=\"M18.56 5.44l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94zm-11 1L8.5 8.5l.94-2.06 2.06-.94-2.06-.94L8.5 2.5l-.94 2.06-2.06.94zm10 10l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94z\"/><path d=\"M17.41 7.96l-1.37-1.37c-.4-.4-.92-.59-1.43-.59-.52 0-1.04.2-1.43.59L10.3 9.45l-7.72 7.72c-.78.78-.78 2.05 0 2.83L4 21.41c.39.39.9.59 1.41.59.51 0 1.02-.2 1.41-.59l7.78-7.78 2.81-2.81c.8-.78.8-2.07 0-2.86zM5.41 20L4 18.59l7.72-7.72 1.47 1.35L5.41 20z\"/>\n",
              "  </svg>\n",
              "      </button>\n",
              "      \n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      flex-wrap:wrap;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "      <script>\n",
              "        const buttonEl =\n",
              "          document.querySelector('#df-d671c676-4248-4715-be15-c885b3e422f9 button.colab-df-convert');\n",
              "        buttonEl.style.display =\n",
              "          google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "        async function convertToInteractive(key) {\n",
              "          const element = document.querySelector('#df-d671c676-4248-4715-be15-c885b3e422f9');\n",
              "          const dataTable =\n",
              "            await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                     [key], {});\n",
              "          if (!dataTable) return;\n",
              "\n",
              "          const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "            '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "            + ' to learn more about interactive tables.';\n",
              "          element.innerHTML = '';\n",
              "          dataTable['output_type'] = 'display_data';\n",
              "          await google.colab.output.renderOutput(dataTable, element);\n",
              "          const docLink = document.createElement('div');\n",
              "          docLink.innerHTML = docLinkHtml;\n",
              "          element.appendChild(docLink);\n",
              "        }\n",
              "      </script>\n",
              "    </div>\n",
              "  </div>\n",
              "  "
            ]
          },
          "metadata": {},
          "execution_count": 32
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "table.to_csv('table.csv')"
      ],
      "metadata": {
        "id": "aVXri30fiL-x"
      },
      "execution_count": 33,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "TmfJKT0rmLyp"
      },
      "source": [
        "ÐŸÑ€ÐµÐ¾Ð±Ñ€Ð°Ð·Ð¾Ð²Ð°Ð½Ð¸Ðµ .sam Ð² .bam (Ð½Ðµ Ñ‚Ñ€ÐµÐ±ÑƒÐµÑ‚ÑÑ ÐµÑÐ»Ð¸ Ð·Ð°Ð¿ÑƒÑÐºÐ°Ñ‚ÑŒ macs2 Ð½Ð° .sam Ñ„Ð°Ð¹Ð»Ð°Ñ…)"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": null,
      "metadata": {
        "id": "2PteJ-KVqFhs"
      },
      "outputs": [],
      "source": [
        "!samtools view -h -S -b -o bowtie2_res/res_VLD.bam bowtie2_res/res_VLD.sam\n",
        "!samtools view -h -S -b -o bowtie2_res/res_SSR.bam bowtie2_res/res_SSR.sam\n",
        "!samtools view -h -S -b -o bowtie2_res/res_YSG.bam bowtie2_res/res_YSG.sam"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "eLtG7xtRup19"
      },
      "source": [
        "ÐÐ° ÑÑ‚Ð¾Ð¼ Ð¼Ð¾Ð¼ÐµÐ½Ñ‚Ðµ Ð¼Ð¾Ð¶Ð½Ð¾ ÑÐºÐ°Ñ‡Ð°Ñ‚ÑŒ ÑÐµÐ±Ðµ Ð½Ð° ÐºÐ¾Ð¼Ð¿ÑŒÑŽÑ‚ÐµÑ€ Ð²ÑÐµ Ð¿Ð¾Ð»ÑƒÑ‡ÐµÐ½Ð½Ñ‹Ðµ .bam Ð¸Ð»Ð¸ .sam Ñ„Ð°Ð¹Ð»Ñ‹ Ð½Ð° ÑÐ»ÑƒÑ‡Ð°Ð¹ ÑÐ±Ð¾Ñ."
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "XZtQrPZMmgnC"
      },
      "source": [
        "## Peak calling\n",
        "\n",
        "ÐŸÐ¾Ð¼Ð¸Ð¼Ð¾ macs2 Ð¼Ð¾Ð¶ÐµÑ‚ Ð±Ñ‹Ñ‚ÑŒ Ð¸ÑÐ¿Ð¾Ð»ÑŒÐ·Ð¾Ð²Ð°Ð½ Ð¿Ð°ÐºÐµÑ‚ Homer Ð´Ð»Ñ Ð¿Ð¾Ð¸ÑÐºÐ° Ð¿Ð¸ÐºÐ¾Ð², Ð° Ñ‚Ð°ÐºÐ¶Ðµ macs3."
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 34,
      "metadata": {
        "id": "uX1ps4VDp_Ae"
      },
      "outputs": [],
      "source": [
        "!mkdir macs2"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 35,
      "metadata": {
        "id": "EJQOCcv2nJW_",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "dec1884b-29b1-4863-ed56-14154f3420f8"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "INFO  @ Wed, 01 Mar 2023 20:08:09: \n",
            "# Command line: callpeak --broad -t bowtie2_res/res_VLD.sam -c bowtie2_res/res_YSG.sam -f SAM --outdir macs2 -n ENCFF815VLD\n",
            "# ARGUMENTS LIST:\n",
            "# name = ENCFF815VLD\n",
            "# format = SAM\n",
            "# ChIP-seq file = ['bowtie2_res/res_VLD.sam']\n",
            "# control file = ['bowtie2_res/res_YSG.sam']\n",
            "# effective genome size = 2.70e+09\n",
            "# band width = 300\n",
            "# model fold = [5, 50]\n",
            "# qvalue cutoff for narrow/strong regions = 5.00e-02\n",
            "# qvalue cutoff for broad/weak regions = 1.00e-01\n",
            "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
            "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
            "# Larger dataset will be scaled towards smaller dataset.\n",
            "# Range for calculating regional lambda is: 1000 bps and 10000 bps\n",
            "# Broad region calling is on\n",
            "# Paired-End mode is off\n",
            " \n",
            "INFO  @ Wed, 01 Mar 2023 20:08:09: #1 read tag files... \n",
            "INFO  @ Wed, 01 Mar 2023 20:08:09: #1 read treatment tags... \n",
            "INFO  @ Wed, 01 Mar 2023 20:08:20:  6000000 \n",
            "INFO  @ Wed, 01 Mar 2023 20:08:23:  8000000 \n",
            "INFO  @ Wed, 01 Mar 2023 20:08:24:  9000000 \n",
            "INFO  @ Wed, 01 Mar 2023 20:08:28:  12000000 \n",
            "INFO  @ Wed, 01 Mar 2023 20:08:30:  14000000 \n",
            "INFO  @ Wed, 01 Mar 2023 20:08:32:  16000000 \n",
            "INFO  @ Wed, 01 Mar 2023 20:08:35:  18000000 \n",
            "INFO  @ Wed, 01 Mar 2023 20:08:56: #1.2 read input tags... \n",
            "INFO  @ Wed, 01 Mar 2023 20:09:13: #1 tag size is determined as 36 bps \n",
            "INFO  @ Wed, 01 Mar 2023 20:09:13: #1 tag size = 36.0 \n",
            "INFO  @ Wed, 01 Mar 2023 20:09:13: #1  total tags in treatment: 4162752 \n",
            "INFO  @ Wed, 01 Mar 2023 20:09:13: #1 user defined the maximum tags... \n",
            "INFO  @ Wed, 01 Mar 2023 20:09:13: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
            "INFO  @ Wed, 01 Mar 2023 20:09:13: #1  tags after filtering in treatment: 3336018 \n",
            "INFO  @ Wed, 01 Mar 2023 20:09:13: #1  Redundant rate of treatment: 0.20 \n",
            "INFO  @ Wed, 01 Mar 2023 20:09:13: #1  total tags in control: 1297244 \n",
            "INFO  @ Wed, 01 Mar 2023 20:09:13: #1 user defined the maximum tags... \n",
            "INFO  @ Wed, 01 Mar 2023 20:09:13: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
            "INFO  @ Wed, 01 Mar 2023 20:09:13: #1  tags after filtering in control: 1081144 \n",
            "INFO  @ Wed, 01 Mar 2023 20:09:13: #1  Redundant rate of control: 0.17 \n",
            "INFO  @ Wed, 01 Mar 2023 20:09:13: #1 finished! \n",
            "INFO  @ Wed, 01 Mar 2023 20:09:13: #2 Build Peak Model... \n",
            "INFO  @ Wed, 01 Mar 2023 20:09:13: #2 looking for paired plus/minus strand peaks... \n",
            "INFO  @ Wed, 01 Mar 2023 20:09:14: #2 number of paired peaks: 24503 \n",
            "INFO  @ Wed, 01 Mar 2023 20:09:14: start model_add_line... \n",
            "INFO  @ Wed, 01 Mar 2023 20:09:14: start X-correlation... \n",
            "INFO  @ Wed, 01 Mar 2023 20:09:14: end of X-cor \n",
            "INFO  @ Wed, 01 Mar 2023 20:09:14: #2 finished! \n",
            "INFO  @ Wed, 01 Mar 2023 20:09:14: #2 predicted fragment length is 36 bps \n",
            "INFO  @ Wed, 01 Mar 2023 20:09:14: #2 alternative fragment length(s) may be 36 bps \n",
            "INFO  @ Wed, 01 Mar 2023 20:09:14: #2.2 Generate R script for model : macs2/ENCFF815VLD_model.r \n",
            "WARNING @ Wed, 01 Mar 2023 20:09:14: #2 Since the d (36) calculated from paired-peaks are smaller than 2*tag length, it may be influenced by unknown sequencing problem! \n",
            "WARNING @ Wed, 01 Mar 2023 20:09:14: #2 You may need to consider one of the other alternative d(s): 36 \n",
            "WARNING @ Wed, 01 Mar 2023 20:09:14: #2 You can restart the process with --nomodel --extsize XXX with your choice or an arbitrary number. Nontheless, MACS will continute computing. \n",
            "INFO  @ Wed, 01 Mar 2023 20:09:14: #3 Call peaks... \n",
            "INFO  @ Wed, 01 Mar 2023 20:09:14: #3 Call broad peaks with given level1 -log10qvalue cutoff and level2: 1.301030, 1.000000... \n",
            "INFO  @ Wed, 01 Mar 2023 20:09:14: #3 Pre-compute pvalue-qvalue table... \n",
            "INFO  @ Wed, 01 Mar 2023 20:09:25: #3 Call peaks for each chromosome... \n",
            "INFO  @ Wed, 01 Mar 2023 20:09:32: #4 Write output xls file... macs2/ENCFF815VLD_peaks.xls \n",
            "INFO  @ Wed, 01 Mar 2023 20:09:32: #4 Write broad peak in broadPeak format file... macs2/ENCFF815VLD_peaks.broadPeak \n",
            "INFO  @ Wed, 01 Mar 2023 20:09:32: #4 Write broad peak in bed12/gappedPeak format file... macs2/ENCFF815VLD_peaks.gappedPeak \n",
            "INFO  @ Wed, 01 Mar 2023 20:09:32: Done! \n"
          ]
        }
      ],
      "source": [
        "!macs2 callpeak --broad -t bowtie2_res/res_VLD.sam \\\n",
        "    -c bowtie2_res/res_YSG.sam \\\n",
        " \t  -f SAM \\\n",
        "\t  --outdir macs2 \\\n",
        "\t\t\t-n ENCFF815VLD"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!macs2 callpeak --broad -t bowtie2_res/res_SSR.sam \\\n",
        "    -c bowtie2_res/res_YSG.sam \\\n",
        " \t  -f SAM \\\n",
        "\t  --outdir macs2 \\\n",
        "\t\t  -n ENCFF644SSR"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "3LqqG8C8kvR1",
        "outputId": "637655a2-73d1-4d1b-de1b-092eac1c4f18"
      },
      "execution_count": 36,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "INFO  @ Wed, 01 Mar 2023 20:09:38: \n",
            "# Command line: callpeak --broad -t bowtie2_res/res_SSR.sam -c bowtie2_res/res_YSG.sam -f SAM --outdir macs2 -n ENCFF644SSR\n",
            "# ARGUMENTS LIST:\n",
            "# name = ENCFF644SSR\n",
            "# format = SAM\n",
            "# ChIP-seq file = ['bowtie2_res/res_SSR.sam']\n",
            "# control file = ['bowtie2_res/res_YSG.sam']\n",
            "# effective genome size = 2.70e+09\n",
            "# band width = 300\n",
            "# model fold = [5, 50]\n",
            "# qvalue cutoff for narrow/strong regions = 5.00e-02\n",
            "# qvalue cutoff for broad/weak regions = 1.00e-01\n",
            "# The maximum gap between significant sites is assigned as the read length/tag size.\n",
            "# The minimum length of peaks is assigned as the predicted fragment length \"d\".\n",
            "# Larger dataset will be scaled towards smaller dataset.\n",
            "# Range for calculating regional lambda is: 1000 bps and 10000 bps\n",
            "# Broad region calling is on\n",
            "# Paired-End mode is off\n",
            " \n",
            "INFO  @ Wed, 01 Mar 2023 20:09:38: #1 read tag files... \n",
            "INFO  @ Wed, 01 Mar 2023 20:09:38: #1 read treatment tags... \n",
            "INFO  @ Wed, 01 Mar 2023 20:09:52:  10000000 \n",
            "INFO  @ Wed, 01 Mar 2023 20:10:18:  29000000 \n",
            "INFO  @ Wed, 01 Mar 2023 20:10:41:  48000000 \n",
            "INFO  @ Wed, 01 Mar 2023 20:10:46:  53000000 \n",
            "INFO  @ Wed, 01 Mar 2023 20:10:51:  57000000 \n",
            "INFO  @ Wed, 01 Mar 2023 20:10:59:  61000000 \n",
            "INFO  @ Wed, 01 Mar 2023 20:11:03:  63000000 \n",
            "INFO  @ Wed, 01 Mar 2023 20:11:04: #1.2 read input tags... \n",
            "INFO  @ Wed, 01 Mar 2023 20:11:17: #1 tag size is determined as 36 bps \n",
            "INFO  @ Wed, 01 Mar 2023 20:11:17: #1 tag size = 36.0 \n",
            "INFO  @ Wed, 01 Mar 2023 20:11:17: #1  total tags in treatment: 8567159 \n",
            "INFO  @ Wed, 01 Mar 2023 20:11:17: #1 user defined the maximum tags... \n",
            "INFO  @ Wed, 01 Mar 2023 20:11:17: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
            "INFO  @ Wed, 01 Mar 2023 20:11:17: #1  tags after filtering in treatment: 5990456 \n",
            "INFO  @ Wed, 01 Mar 2023 20:11:17: #1  Redundant rate of treatment: 0.30 \n",
            "INFO  @ Wed, 01 Mar 2023 20:11:17: #1  total tags in control: 1297244 \n",
            "INFO  @ Wed, 01 Mar 2023 20:11:17: #1 user defined the maximum tags... \n",
            "INFO  @ Wed, 01 Mar 2023 20:11:17: #1 filter out redundant tags at the same location and the same strand by allowing at most 1 tag(s) \n",
            "INFO  @ Wed, 01 Mar 2023 20:11:17: #1  tags after filtering in control: 1081144 \n",
            "INFO  @ Wed, 01 Mar 2023 20:11:17: #1  Redundant rate of control: 0.17 \n",
            "INFO  @ Wed, 01 Mar 2023 20:11:17: #1 finished! \n",
            "INFO  @ Wed, 01 Mar 2023 20:11:17: #2 Build Peak Model... \n",
            "INFO  @ Wed, 01 Mar 2023 20:11:17: #2 looking for paired plus/minus strand peaks... \n",
            "INFO  @ Wed, 01 Mar 2023 20:11:20: #2 number of paired peaks: 32805 \n",
            "INFO  @ Wed, 01 Mar 2023 20:11:20: start model_add_line... \n",
            "INFO  @ Wed, 01 Mar 2023 20:11:21: start X-correlation... \n",
            "INFO  @ Wed, 01 Mar 2023 20:11:21: end of X-cor \n",
            "INFO  @ Wed, 01 Mar 2023 20:11:21: #2 finished! \n",
            "INFO  @ Wed, 01 Mar 2023 20:11:21: #2 predicted fragment length is 36 bps \n",
            "INFO  @ Wed, 01 Mar 2023 20:11:21: #2 alternative fragment length(s) may be 36 bps \n",
            "INFO  @ Wed, 01 Mar 2023 20:11:21: #2.2 Generate R script for model : macs2/ENCFF644SSR_model.r \n",
            "WARNING @ Wed, 01 Mar 2023 20:11:21: #2 Since the d (36) calculated from paired-peaks are smaller than 2*tag length, it may be influenced by unknown sequencing problem! \n",
            "WARNING @ Wed, 01 Mar 2023 20:11:21: #2 You may need to consider one of the other alternative d(s): 36 \n",
            "WARNING @ Wed, 01 Mar 2023 20:11:21: #2 You can restart the process with --nomodel --extsize XXX with your choice or an arbitrary number. Nontheless, MACS will continute computing. \n",
            "INFO  @ Wed, 01 Mar 2023 20:11:21: #3 Call peaks... \n",
            "INFO  @ Wed, 01 Mar 2023 20:11:21: #3 Call broad peaks with given level1 -log10qvalue cutoff and level2: 1.301030, 1.000000... \n",
            "INFO  @ Wed, 01 Mar 2023 20:11:21: #3 Pre-compute pvalue-qvalue table... \n",
            "INFO  @ Wed, 01 Mar 2023 20:11:33: #3 Call peaks for each chromosome... \n",
            "INFO  @ Wed, 01 Mar 2023 20:11:38: #4 Write output xls file... macs2/ENCFF644SSR_peaks.xls \n",
            "INFO  @ Wed, 01 Mar 2023 20:11:38: #4 Write broad peak in broadPeak format file... macs2/ENCFF644SSR_peaks.broadPeak \n",
            "INFO  @ Wed, 01 Mar 2023 20:11:38: #4 Write broad peak in bed12/gappedPeak format file... macs2/ENCFF644SSR_peaks.gappedPeak \n",
            "INFO  @ Wed, 01 Mar 2023 20:11:38: Done! \n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 37,
      "metadata": {
        "id": "HrNwxs7ArPz7",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "434364a7-d6d1-4eba-ceeb-629b2c6ebaa4"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2023-03-01 20:11:42--  https://www.encodeproject.org/files/ENCFF917LDY/@@download/ENCFF917LDY.bed.gz\n",
            "Resolving www.encodeproject.org (www.encodeproject.org)... 34.211.244.144\n",
            "Connecting to www.encodeproject.org (www.encodeproject.org)|34.211.244.144|:443... connected.\n",
            "HTTP request sent, awaiting response... 307 Temporary Redirect\n",
            "Location: https://encode-public.s3.amazonaws.com/2016/11/09/6bc0f5b3-e514-4c04-a4e4-45f23dfc77b3/ENCFF917LDY.bed.gz?response-content-disposition=attachment%3B%20filename%3DENCFF917LDY.bed.gz&AWSAccessKeyId=ASIATGZNGCNXQ5WMEO4C&Signature=UqVzetswvldRt34wqlMs6GfU9Dc%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEOz%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEaCXVzLXdlc3QtMiJHMEUCIQCiNA%2FRwooqWVfF%2BTRUnyXZ5BGjFYm2fWX54t7OiSozGAIgIGe9QCDR6sQ%2BrNZLzCcIjQuNL4FaAAyDvusN0U6u6kgquwUIlf%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FARAAGgwyMjA3NDg3MTQ4NjMiDKMKJbc75euvCmObtCqPBUt6OtstHkuSVW8oyuN%2BMFIdHEQ3xSdSZNdjOwX6ZXlZgLmezO2EDjS%2F5oVBxVeTSomfnA2g9VSRGec10NdyggkM%2FGWDZdJ45fzgSIkqj3E%2B6OuZcgN8yKABi1RCe4hY5pu3o18cGZrWagjORWdk1LLrYhlrdgOPT4UzuJGgmqTpJ2XxFmi%2FowfE3j9uNAD3wMJnDEsFJnNHWn4t%2Fuom0kW2zWbsKMLjiRRGX4%2BiwvubjUhnsvPhG0Td024CH97MU89OO7yTsDs%2BUAOzwEaIyq%2BmM%2FdGuNu1MLNDS8FMAHhhu84w%2BfVDOczhYG8hjO%2By3Yr%2BSMI%2Fa7aU%2FHMphgKG3W1RB8swN%2Bo6Xw%2Blh06aY%2Bx8cr22kBAbxUCPp3GanxWbThLYO24%2BKW4j9yAQCrQdmLyeLlg%2FHCgGmpwJ9UJvZaklyWpWBgIWA1SBYxZurVz4K8rmJoIRE0kdFdHvSJoafcvYga%2FCr%2FSEK4RCKdPEIsbv6%2FTqxh1hJF5%2F0qtknLQaRTN1%2BZd16JiLDkql20ocu%2Fg0%2BQeUB4BQGiClQ5wyPPb%2FpvkcWqwx3y9DfPVnrpceDkRqH67ABMKzaNf5ppntwzwWjHHtJvP3JDNSuyJ3%2Bims79P%2Bx%2BYa3Ng%2BT8bzJQjduqNQpDqHcNXf21LCKFjspKe7mRIrmMsbr%2FK2GOZyGZjlFlA1wdVpMP2wq%2Fsz6uqEmBRSJERJczR8ohzXv5Ili9boT438hcSJtxz2qtp8z046qUPhHfL5LkQ1VKb6E0PjzBKgeDGtVG9mDqc26lEO6h0GQgnCHRU8oq3QG0DlJHu53J5pV6Nh20Q7BONBB1e53aZ%2F8L23HNcTGG42obRgv1v%2B5Mvs%2BOnjdjYj9xYrD8ow4Nf%2BnwY6sQGX7C1JSgkkeBULlfoOSlGfViwF8PPieVxtYERbl1y2I8hXL6%2FNx%2BYXFUsSfs74k5BXKEf1fBFvPedZC%2FlgEwW7jsu9uVqtbenvYRVHKHzxIDCr9GpH8%2BEK%2Fp77yc81EiR0uBQEA0m2rJI7VYyYm%2Fo2b%2BdjQ7LraJCPxRIOqZfY7a8r1QWCycPMmCRy9v0xGdTQ%2FzRa82Ib1JlQfnH8JCbnxfoj6CwfDV0yrhiUsQHiNHQ%3D&Expires=1677831102 [following]\n",
            "--2023-03-01 20:11:42--  https://encode-public.s3.amazonaws.com/2016/11/09/6bc0f5b3-e514-4c04-a4e4-45f23dfc77b3/ENCFF917LDY.bed.gz?response-content-disposition=attachment%3B%20filename%3DENCFF917LDY.bed.gz&AWSAccessKeyId=ASIATGZNGCNXQ5WMEO4C&Signature=UqVzetswvldRt34wqlMs6GfU9Dc%3D&x-amz-security-token=IQoJb3JpZ2luX2VjEOz%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FwEaCXVzLXdlc3QtMiJHMEUCIQCiNA%2FRwooqWVfF%2BTRUnyXZ5BGjFYm2fWX54t7OiSozGAIgIGe9QCDR6sQ%2BrNZLzCcIjQuNL4FaAAyDvusN0U6u6kgquwUIlf%2F%2F%2F%2F%2F%2F%2F%2F%2F%2FARAAGgwyMjA3NDg3MTQ4NjMiDKMKJbc75euvCmObtCqPBUt6OtstHkuSVW8oyuN%2BMFIdHEQ3xSdSZNdjOwX6ZXlZgLmezO2EDjS%2F5oVBxVeTSomfnA2g9VSRGec10NdyggkM%2FGWDZdJ45fzgSIkqj3E%2B6OuZcgN8yKABi1RCe4hY5pu3o18cGZrWagjORWdk1LLrYhlrdgOPT4UzuJGgmqTpJ2XxFmi%2FowfE3j9uNAD3wMJnDEsFJnNHWn4t%2Fuom0kW2zWbsKMLjiRRGX4%2BiwvubjUhnsvPhG0Td024CH97MU89OO7yTsDs%2BUAOzwEaIyq%2BmM%2FdGuNu1MLNDS8FMAHhhu84w%2BfVDOczhYG8hjO%2By3Yr%2BSMI%2Fa7aU%2FHMphgKG3W1RB8swN%2Bo6Xw%2Blh06aY%2Bx8cr22kBAbxUCPp3GanxWbThLYO24%2BKW4j9yAQCrQdmLyeLlg%2FHCgGmpwJ9UJvZaklyWpWBgIWA1SBYxZurVz4K8rmJoIRE0kdFdHvSJoafcvYga%2FCr%2FSEK4RCKdPEIsbv6%2FTqxh1hJF5%2F0qtknLQaRTN1%2BZd16JiLDkql20ocu%2Fg0%2BQeUB4BQGiClQ5wyPPb%2FpvkcWqwx3y9DfPVnrpceDkRqH67ABMKzaNf5ppntwzwWjHHtJvP3JDNSuyJ3%2Bims79P%2Bx%2BYa3Ng%2BT8bzJQjduqNQpDqHcNXf21LCKFjspKe7mRIrmMsbr%2FK2GOZyGZjlFlA1wdVpMP2wq%2Fsz6uqEmBRSJERJczR8ohzXv5Ili9boT438hcSJtxz2qtp8z046qUPhHfL5LkQ1VKb6E0PjzBKgeDGtVG9mDqc26lEO6h0GQgnCHRU8oq3QG0DlJHu53J5pV6Nh20Q7BONBB1e53aZ%2F8L23HNcTGG42obRgv1v%2B5Mvs%2BOnjdjYj9xYrD8ow4Nf%2BnwY6sQGX7C1JSgkkeBULlfoOSlGfViwF8PPieVxtYERbl1y2I8hXL6%2FNx%2BYXFUsSfs74k5BXKEf1fBFvPedZC%2FlgEwW7jsu9uVqtbenvYRVHKHzxIDCr9GpH8%2BEK%2Fp77yc81EiR0uBQEA0m2rJI7VYyYm%2Fo2b%2BdjQ7LraJCPxRIOqZfY7a8r1QWCycPMmCRy9v0xGdTQ%2FzRa82Ib1JlQfnH8JCbnxfoj6CwfDV0yrhiUsQHiNHQ%3D&Expires=1677831102\n",
            "Resolving encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)... 52.92.211.161, 52.218.224.82, 52.92.133.49, ...\n",
            "Connecting to encode-public.s3.amazonaws.com (encode-public.s3.amazonaws.com)|52.92.211.161|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 1567219 (1.5M) [binary/octet-stream]\n",
            "Saving to: â€˜ENCFF917LDY.bed.gzâ€™\n",
            "\n",
            "ENCFF917LDY.bed.gz  100%[===================>]   1.49M  6.08MB/s    in 0.2s    \n",
            "\n",
            "2023-03-01 20:11:43 (6.08 MB/s) - â€˜ENCFF917LDY.bed.gzâ€™ saved [1567219/1567219]\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!wget https://www.encodeproject.org/files/ENCFF917LDY/@@download/ENCFF917LDY.bed.gz\n",
        "!gzip -d ENCFF917LDY.bed.gz"
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "MYEGEbQjvEwY"
      },
      "source": [
        "## Ð¡Ñ€Ð°Ð²Ð½ÐµÐ½Ð¸Ðµ Ñ€ÐµÐ·ÑƒÐ»ÑŒÑ‚Ð°Ñ‚Ð¾Ð²\n",
        "\n",
        "Ð¡Ñ€Ð°Ð²Ð½Ð¸Ð²Ð°ÐµÐ¼ Ñ‚Ðµ Ð¿Ð¸ÐºÐ¸, ÐºÐ¾Ñ‚Ð¾Ñ€Ñ‹Ðµ Ð¼Ñ‹ Ð¿Ð¾Ð»ÑƒÑ‡Ð¸Ð»Ð¸, Ñ Ð¿Ð¸ÐºÐ°Ð¼Ð¸, ÐºÐ¾Ñ‚Ð¾Ñ€Ñ‹Ðµ Ð¿Ñ€Ð¸Ð²ÐµÐ´ÐµÐ½Ñ‹ Ð² ENCODE (Ð²Ð°Ð¶Ð½Ð¾, Ñ‡Ñ‚Ð¾Ð±Ñ‹ Ð²ÐµÑ€ÑÐ¸Ð¸ Ð³ÐµÐ½Ð¾Ð¼Ð° hg38 Ð¸Ð»Ð¸ hg19) Ð´Ð»Ñ .bed Ñ„Ð°Ð¹Ð»Ð° Ð¸Ð· ENCODE Ð¸ Ñ‚Ð¾Ð¹ Ñ…Ñ€Ð¾Ð¼Ð¾ÑÐ¾Ð¼Ñ‹, ÐºÐ¾Ñ‚Ð¾Ñ€ÑƒÑŽ ÑÐºÐ°Ñ‡Ð¸Ð²Ð°Ð»Ð¸ Ð²Ñ‹ÑˆÐµ, ÑÐ¾Ð¿Ð°Ð´Ð°Ð»Ð¸.\n",
        "\n",
        "ÐŸÑ€Ð¾Ð°Ð½Ð°Ð»Ð¸Ð·Ð¸Ñ€ÑƒÐ¹Ñ‚Ðµ Ð¿Ð¾Ð»ÑƒÑ‡ÐµÐ½Ð½Ñ‹Ðµ Ñ€ÐµÐ·ÑƒÐ»ÑŒÑ‚Ð°Ñ‚Ñ‹ Ð¸ Ð¿Ñ€Ð¸Ð²ÐµÐ´Ð¸Ñ‚Ðµ ÑÐ²Ð¾Ð¸ Ñ€Ð°ÑÑÑƒÐ¶Ð´ÐµÐ½Ð¸Ñ Ð² README.md. ÐšÐ°Ðº Ð¼Ð¾Ð¶Ð½Ð¾ Ð¾Ð±ÑŠÑÑÐ½Ð¸Ñ‚ÑŒ Ñ€Ð°Ð·Ð»Ð¸Ñ‡Ð¸Ñ Ð² ÐºÐ¾Ð»Ð¸Ñ‡ÐµÑÑ‚Ð²Ðµ Ð¿ÐµÑ€ÐµÑÐµÑ‡ÐµÐ½Ð¸Ð¹?"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!mkdir venn1\n",
        "!mkdir venn2\n",
        "!mkdir venn3\n",
        "!mkdir venn4"
      ],
      "metadata": {
        "id": "Oi4RHDgrqB2l"
      },
      "execution_count": 38,
      "outputs": []
    },
    {
      "cell_type": "code",
      "execution_count": 39,
      "metadata": {
        "id": "_1EAygSCvc8a",
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "outputId": "507fc29c-40a0-4a3c-83ea-9321e0aa6795"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "Generating a 2-way \"venn\" diagram. Please wait...\n",
            "\n",
            "\n",
            "Done! Please check your results @ venn1. \n",
            "Thank you for using Intervene!\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!intervene venn -i macs2/ENCFF815VLD_peaks.broadPeak ENCFF917LDY.bed -o venn1 --filenames"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!intervene venn -i ENCFF917LDY.bed macs2/ENCFF815VLD_peaks.broadPeak -o venn2 --filenames"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "MOzVfolnqYes",
        "outputId": "8fc49a18-fee8-44b5-8180-9154aa8db89f"
      },
      "execution_count": 40,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "Generating a 2-way \"venn\" diagram. Please wait...\n",
            "\n",
            "\n",
            "Done! Please check your results @ venn2. \n",
            "Thank you for using Intervene!\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!intervene venn -i macs2/ENCFF644SSR_peaks.broadPeak ENCFF917LDY.bed -o venn3 --filenames"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "76qK-3LaqYj1",
        "outputId": "b6a69398-eaa4-43ed-eed3-54cd4c3743a7"
      },
      "execution_count": 41,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "Generating a 2-way \"venn\" diagram. Please wait...\n",
            "\n",
            "\n",
            "Done! Please check your results @ venn3. \n",
            "Thank you for using Intervene!\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!intervene venn -i ENCFF917LDY.bed macs2/ENCFF644SSR_peaks.broadPeak -o venn4 --filenames"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "IxoyruMnqi2c",
        "outputId": "2abe73ad-18b0-4c5d-f309-a56074fd4f9e"
      },
      "execution_count": 42,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "Generating a 2-way \"venn\" diagram. Please wait...\n",
            "\n",
            "\n",
            "Done! Please check your results @ venn4. \n",
            "Thank you for using Intervene!\n",
            "\n"
          ]
        }
      ]
    }
  ],
  "metadata": {
    "colab": {
      "provenance": []
    },
    "kernelspec": {
      "display_name": "Python 3",
      "name": "python3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "nbformat": 4,
  "nbformat_minor": 0
}